--- src/base/unix_socket.cc
+++ src/base/unix_socket.cc
@@ -62,7 +62,7 @@
 #include "perfetto/ext/base/string_utils.h"
 #include "perfetto/ext/base/utils.h"
 
-#if PERFETTO_BUILDFLAG(PERFETTO_OS_LINUX) || \
+#if PERFETTO_BUILDFLAG(PERFETTO_OS_LINUX_BUT_NOT_QNX) || \
     PERFETTO_BUILDFLAG(PERFETTO_OS_ANDROID)
 // Use a local stripped copy of vm_sockets.h from UAPI.
 #include "src/base/vm_sockets.h"
@@ -91,7 +91,7 @@
 using CBufLenType = socklen_t;
 #endif
 
-#if PERFETTO_BUILDFLAG(PERFETTO_OS_LINUX) || \
+#if PERFETTO_BUILDFLAG(PERFETTO_OS_LINUX_BUT_NOT_QNX) || \
     PERFETTO_BUILDFLAG(PERFETTO_OS_ANDROID)
 constexpr char kVsockNamePrefix[] = "vsock://";
 #endif
@@ -207,7 +207,7 @@
       SockaddrAny res(addr_info->ai_addr,
                       static_cast<socklen_t>(addr_info->ai_addrlen));
       freeaddrinfo(addr_info);
-      return res;
+      return std::move(res);
     }
     case SockFamily::kInet6: {
       auto parts = SplitString(socket_name, "]");
@@ -225,10 +225,10 @@
       SockaddrAny res(addr_info->ai_addr,
                       static_cast<socklen_t>(addr_info->ai_addrlen));
       freeaddrinfo(addr_info);
-      return res;
+      return std::move(res);
     }
     case SockFamily::kVsock: {
-#if PERFETTO_BUILDFLAG(PERFETTO_OS_LINUX) || \
+#if PERFETTO_BUILDFLAG(PERFETTO_OS_LINUX_BUT_NOT_QNX) || \
     PERFETTO_BUILDFLAG(PERFETTO_OS_ANDROID)
       PERFETTO_CHECK(StartsWith(socket_name, kVsockNamePrefix));
       auto address_port = StripPrefix(socket_name, kVsockNamePrefix);
@@ -246,8 +246,7 @@
         addr.svm_flags = VMADDR_FLAG_TO_HOST;
       }
 #endif
-      SockaddrAny res(&addr, sizeof(addr));
-      return res;
+      return SockaddrAny(&addr, sizeof(addr));
 #else
       errno = ENOTSOCK;
       return SockaddrAny();
@@ -328,7 +327,7 @@
   if (addr[0] == '@')
     return SockFamily::kUnix;  // Abstract AF_UNIX sockets.
 
-#if PERFETTO_BUILDFLAG(PERFETTO_OS_LINUX) || \
+#if PERFETTO_BUILDFLAG(PERFETTO_OS_LINUX_BUT_NOT_QNX) || \
     PERFETTO_BUILDFLAG(PERFETTO_OS_ANDROID)
   // Vsock address starts with vsock://.
   if (strncmp(addr, kVsockNamePrefix, strlen(kVsockNamePrefix)) == 0)
@@ -690,7 +689,9 @@
                             size_t num_fds) {
   PERFETTO_DCHECK(fd_);
   msghdr msg_hdr = {};
-  iovec iov = {const_cast<void*>(msg), len};
+  iovec iov = {};
+  iov.iov_base = const_cast<void*>(msg);
+  iov.iov_len = len;
   msg_hdr.msg_iov = &iov;
   msg_hdr.msg_iovlen = 1;
   alignas(cmsghdr) char control_buf[256];
@@ -721,7 +722,9 @@
                                size_t max_files) {
   PERFETTO_DCHECK(fd_);
   msghdr msg_hdr = {};
-  iovec iov = {msg, len};
+  iovec iov = {};
+  iov.iov_base = msg;
+  iov.iov_len = len;
   msg_hdr.msg_iov = &iov;
   msg_hdr.msg_iovlen = 1;
   alignas(cmsghdr) char control_buf[256];
--- src/base/vm_sockets.h
+++ src/base/vm_sockets.h
@@ -19,18 +19,15 @@
 
 #include "perfetto/base/build_config.h"
 
-#if PERFETTO_BUILDFLAG(PERFETTO_OS_LINUX) || \
+#if PERFETTO_BUILDFLAG(PERFETTO_OS_LINUX_BUT_NOT_QNX) || \
     PERFETTO_BUILDFLAG(PERFETTO_OS_ANDROID)
 
 #include <sys/socket.h>
 
-#if PERFETTO_BUILDFLAG(PERFETTO_OS_QNX)
-// Requires the QNX Advanced Virtualization Framework
-#include <vm_sockets.h>
-#elif defined(AF_VSOCK)
+#if defined(AF_VSOCK)
 // Use system vm_socket.h if available.
 #include <linux/vm_sockets.h>
-#else  // defined(AF_SOCK)
+#else  // !defined(AF_VSOCK)
 // Fallback and use the stripped copy from the UAPI vm_sockets.h.
 
 #include <stdint.h>  // For uint8_t.
@@ -48,9 +45,9 @@
                          sizeof(unsigned int) - sizeof(uint8_t)];
 };
 
-#endif  // defined(AF_SOCK)
+#endif  // !defined(AF_VSOCK)
 
-#endif  // PERFETTO_BUILDFLAG(PERFETTO_OS_LINUX) ||
+#endif  // PERFETTO_BUILDFLAG(PERFETTO_OS_LINUX_BUT_NOT_QNX) ||
         // PERFETTO_BUILDFLAG(PERFETTO_OS_ANDROID)
 
 #endif  // SRC_BASE_VM_SOCKETS_H_
--- include/perfetto/ext/base/unix_socket.h
+++ include/perfetto/ext/base/unix_socket.h
@@ -470,7 +470,7 @@
     !PERFETTO_BUILDFLAG(PERFETTO_OS_FUCHSIA)
   uid_t peer_uid_ = kInvalidUid;
 #endif
-#if PERFETTO_BUILDFLAG(PERFETTO_OS_LINUX) || \
+#if PERFETTO_BUILDFLAG(PERFETTO_OS_LINUX_BUT_NOT_QNX) || \
     PERFETTO_BUILDFLAG(PERFETTO_OS_ANDROID)
   pid_t peer_pid_ = kInvalidPid;
 #endif
