diff --git a/third_party/perfetto/src/base/test/vm_test_utils.cc b/third_party/perfetto/src/base/test/vm_test_utils.cc
index 16ac2780a1..9b399636e2 100644
--- a/third_party/perfetto/src/base/test/vm_test_utils.cc
+++ b/third_party/perfetto/src/base/test/vm_test_utils.cc
@@ -92,6 +92,10 @@ bool IsMapped(void* start, size_t size) {
   // Fuchsia doesn't yet support paging (b/119503290).
   ignore_result(page_size);
   return true;
+#elif PERFETTO_BUILDFLAG(PERFETTO_OS_QNX)
+  // mincore() is not available on QNX.
+  ignore_result(page_size);
+  return true;
 #elif PERFETTO_BUILDFLAG(PERFETTO_OS_NACL)
   // mincore isn't available on NaCL.
   ignore_result(page_size);
