diff --git chrome/browser/about_flags.cc chrome/browser/about_flags.cc
index 016ba0e25ff2b6dabca957a27bd4953b808b0740..fa779b50106082b00b4fc53be457d4e30fac7f88 100644
--- chrome/browser/about_flags.cc
+++ chrome/browser/about_flags.cc
@@ -1244 +1244 @@ const FeatureEntry::FeatureVariation
-    BUILDFLAG(IS_WIN)
+    BUILDFLAG(IS_WIN) || BUILDFLAG(IS_QNX)
@@ -1245,0 +1246 @@ const FeatureEntry::FeatureVariation
+#if !BUILDFLAG(IS_QNX)
@@ -1578,0 +1580 @@ const FeatureEntry::FeatureVariation kOmniboxToolbeltVariations[] = {
+#endif  // !BUILDFLAG(IS_QNX)
@@ -11399,0 +11402 @@ const FeatureEntry kFeatureEntries[] = {
+#if BUILDFLAG(ENABLE_PRINTING)
@@ -11405,0 +11409 @@ const FeatureEntry kFeatureEntries[] = {
+#endif
