diff --git chrome/browser/ui/webui/about/BUILD.gn chrome/browser/ui/webui/about/BUILD.gn
index e5d9b020062c440096152249df0f9b1ecc7d3ec3..d33bad85ff09c735bb84da5da24dc0c243287837 100644
--- chrome/browser/ui/webui/about/BUILD.gn
+++ chrome/browser/ui/webui/about/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos || is_android)
+assert(is_win || is_mac || is_linux || is_chromeos || is_android || is_qnx)
 
 source_set("about") {
   sources = [ "about_ui.h" ]
