diff --git chrome/browser/ui/webui/about/BUILD.gn chrome/browser/ui/webui/about/BUILD.gn
index e5d9b020062c4..d7bd31d52f65f 100644
--- chrome/browser/ui/webui/about/BUILD.gn
+++ chrome/browser/ui/webui/about/BUILD.gn
@@ -2,7 +2,9 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos || is_android)
+import("//cef/libcef/features/features.gni")
+
+assert(is_win || is_mac || is_linux || is_chromeos || is_android || is_qnx)
 
 source_set("about") {
   sources = [ "about_ui.h" ]
@@ -30,6 +32,10 @@ source_set("impl") {
     "//components/webui/about",
   ]
 
+  if (enable_cef) {
+    deps += [ "//cef:cef_resources" ]
+  }
+
   if (is_chromeos) {
     deps += [
       "//chrome/browser/ash/borealis",
