diff --git a/third_party/angle/src/tests/BUILD.gn b/third_party/angle/src/tests/BUILD.gn
index bc9a221e6b..0723636437 100644
--- a/third_party/angle/src/tests/BUILD.gn
+++ b/third_party/angle/src/tests/BUILD.gn
@@ -10,7 +10,7 @@ declare_args() {
   build_angle_gles1_conform_tests = checkout_angle_internal
   build_angle_trace_perf_tests = false
   build_angle_perftests =
-      is_win || is_linux || is_chromeos || is_android || is_apple || is_fuchsia
+      is_win || is_linux || is_qnx || is_chromeos || is_android || is_apple || is_fuchsia
 }
 
 declare_args() {
@@ -272,7 +272,7 @@ if (is_ios) {
   }
 }
 
-if (is_win || is_linux || is_chromeos || is_android || is_fuchsia || is_apple) {
+if (is_win || is_linux || is_qnx || is_chromeos || is_android || is_fuchsia || is_apple) {
   import("angle_end2end_tests.gni")
 
   angle_test("angle_end2end_tests") {
@@ -530,7 +530,7 @@ if (angle_enable_cl_testing) {
   }
 }
 
-if (is_win || is_linux || is_chromeos || is_android || is_fuchsia || is_apple) {
+if (is_win || is_linux || is_qnx || is_chromeos || is_android || is_fuchsia || is_apple) {
   import("angle_white_box_tests.gni")
 
   angle_test("angle_white_box_tests") {
@@ -624,7 +624,7 @@ angle_perftests_common("angle_perftests_static") {
   test_utils = ":angle_common_test_utils_static"
 }
 
-if (is_win || is_linux || is_chromeos || is_android || is_apple) {
+if (is_win || is_linux || is_qnx || is_chromeos || is_android || is_apple) {
   import("angle_perftests.gni")
 
   # This test suite is for perf tests that use ANGLE's internals in some direct way.
@@ -703,7 +703,7 @@ if (build_angle_perftests) {
       data_deps += [ "$angle_root/third_party/mesa" ]
     }
 
-    if (is_win || is_linux || is_chromeos) {
+    if (is_win || is_linux || is_qnx || is_chromeos) {
       data_deps += [ "${angle_root}/third_party/glmark2:glmark2_angle" ]
     }
     if (is_win && target_cpu != "arm64") {
