diff --git src/libANGLE/Context.cpp src/libANGLE/Context.cpp
index 44fb2a4bfe4dfc2319616441214a24b740d0f322..45b8267f6e3e54a68e7fb0ed33beea22f51ed4e2 100644
--- src/libANGLE/Context.cpp
+++ src/libANGLE/Context.cpp
@@ -639,6 +639,29 @@ void SetCurrentValidContextTLS(Context *context)
     ASSERT(CurrentValidContextIndex != TLS_INVALID_INDEX);
     angle::SetTLSValue(CurrentValidContextIndex, context);
 }
+#elif defined(__QNXNTO__)
+static angle::TLSIndex GetCurrentValidContextTLSIndex()
+{
+    static pthread_once_t once = PTHREAD_ONCE_INIT;
+    static angle::TLSIndex CurrentValidContextIndex = TLS_INVALID_INDEX;
+    pthread_once(&once, []() {
+      ASSERT(CurrentValidContextIndex == TLS_INVALID_INDEX);
+      CurrentValidContextIndex = angle::CreateTLSIndex(nullptr);
+    });
+    return CurrentValidContextIndex;
+}
+Context *GetCurrentValidContextTLS()
+{
+    angle::TLSIndex CurrentValidContextIndex = GetCurrentValidContextTLSIndex();
+    ASSERT(CurrentValidContextIndex != TLS_INVALID_INDEX);
+    return static_cast<Context *>(angle::GetTLSValue(CurrentValidContextIndex));
+}
+void SetCurrentValidContextTLS(Context *context)
+{
+    angle::TLSIndex CurrentValidContextIndex = GetCurrentValidContextTLSIndex();
+    ASSERT(CurrentValidContextIndex != TLS_INVALID_INDEX);
+    angle::SetTLSValue(CurrentValidContextIndex, context);
+}
 #elif defined(ANGLE_USE_STATIC_THREAD_LOCAL_VARIABLES)
 static thread_local Context *gCurrentValidContext = nullptr;
 Context *GetCurrentValidContextTLS()
@@ -664,7 +687,8 @@ extern void SetCurrentValidContext(Context *context)
     }
 #endif
 
-#if defined(ANGLE_PLATFORM_APPLE) || defined(ANGLE_USE_STATIC_THREAD_LOCAL_VARIABLES)
+#if defined(ANGLE_PLATFORM_APPLE) || defined(ANGLE_USE_STATIC_THREAD_LOCAL_VARIABLES) || \
+    defined(__QNXNTO__)
     SetCurrentValidContextTLS(context);
 #else
     gCurrentValidContext = context;
diff --git src/libANGLE/Context.h src/libANGLE/Context.h
index b18be15deb54665b605d27ef93e52f41e320cd42..2bca24bc5ee40e257317e41f3c02210faf6a9762 100644
--- src/libANGLE/Context.h
+++ src/libANGLE/Context.h
@@ -1186,7 +1186,8 @@ class [[nodiscard]] ScopedContextRef
 };
 
 // Thread-local current valid context bound to the thread.
-#if defined(ANGLE_PLATFORM_APPLE) || defined(ANGLE_USE_STATIC_THREAD_LOCAL_VARIABLES)
+#if defined(ANGLE_PLATFORM_APPLE) || defined(ANGLE_USE_STATIC_THREAD_LOCAL_VARIABLES) || \
+    defined(__QNXNTO__)
 extern Context *GetCurrentValidContextTLS();
 extern void SetCurrentValidContextTLS(Context *context);
 #else
diff --git src/libANGLE/Display.cpp src/libANGLE/Display.cpp
index 72fb02c02de6db120c03b194fa47f01f9b4bfb49..daf47bbe1b47c5e5ae41f6da28c5b48441870391 100644
--- src/libANGLE/Display.cpp
+++ src/libANGLE/Display.cpp
@@ -132,6 +132,29 @@ void SetDisplayTLS(TLSData *tlsData)
     ASSERT(DisplayIndex != TLS_INVALID_INDEX);
     angle::SetTLSValue(DisplayIndex, tlsData);
 }
+#elif defined(__QNXNTO__)
+static angle::TLSIndex GetDisplayTLSIndex()
+{
+    static pthread_once_t once = PTHREAD_ONCE_INIT;
+    static angle::TLSIndex DisplayIndex = TLS_INVALID_INDEX;
+    pthread_once(&once, []() {
+      ASSERT(DisplayIndex == TLS_INVALID_INDEX);
+      DisplayIndex = angle::CreateTLSIndex(nullptr);
+    });
+    return DisplayIndex;
+}
+TLSData *GetDisplayTLS()
+{
+    angle::TLSIndex DisplayIndex = GetDisplayTLSIndex();
+    ASSERT(DisplayIndex != TLS_INVALID_INDEX);
+    return static_cast<TLSData *>(angle::GetTLSValue(DisplayIndex));
+}
+void SetDisplayTLS(TLSData *tlsData)
+{
+    angle::TLSIndex DisplayIndex = GetDisplayTLSIndex();
+    ASSERT(DisplayIndex != TLS_INVALID_INDEX);
+    angle::SetTLSValue(DisplayIndex, tlsData);
+}
 #else
 // Tail calls generated during execution of the entry point, to be run at the end of the entry
 // point.  gTLSData->unlockedTailCall.run() is called at the end of any EGL entry point that is
@@ -2805,7 +2828,7 @@ void Display::InitTLS()
 {
     TLSData *tlsData = new TLSData;
 
-#if defined(ANGLE_PLATFORM_APPLE)
+#if defined(ANGLE_PLATFORM_APPLE) || defined(__QNXNTO__)
     SetDisplayTLS(tlsData);
 #else
     gDisplayTLS = tlsData;
diff --git src/libGLESv2/global_state.cpp src/libGLESv2/global_state.cpp
index 0a248caa55accf8b493a596628cc105b11a9e53f..9e182cb1623c351b24b5f0074857ccada8485f31 100644
--- src/libGLESv2/global_state.cpp
+++ src/libGLESv2/global_state.cpp
@@ -11,6 +11,9 @@
 #include "common/debug.h"
 #include "common/platform.h"
 #include "common/system_utils.h"
+#if defined(__QNXNTO__)
+#    include "common/terminate_capture_qnx.h"
+#endif
 #include "libANGLE/ErrorStrings.h"
 #include "libANGLE/Thread.h"
 #include "libGLESv2/egl_stubs_autogen.h"
@@ -52,7 +55,8 @@ Thread *AllocateCurrentThread()
         // Display TLS data is also intentionally leaked.
         ANGLE_SCOPED_DISABLE_LSAN();
         thread = new Thread();
-#if defined(ANGLE_PLATFORM_APPLE) || defined(ANGLE_USE_STATIC_THREAD_LOCAL_VARIABLES)
+#if defined(ANGLE_PLATFORM_APPLE) || defined(ANGLE_USE_STATIC_THREAD_LOCAL_VARIABLES) || \
+    defined(__QNXNTO__)
         SetCurrentThreadTLS(thread);
 #else
         gCurrentThread = thread;
@@ -112,6 +116,30 @@ void SetCurrentThreadTLS(Thread *thread)
     ASSERT(CurrentThreadIndex != TLS_INVALID_INDEX);
     angle::SetTLSValue(CurrentThreadIndex, thread);
 }
+#elif defined(__QNXNTO__)
+static angle::TLSIndex GetCurrentThreadTLSIndex()
+{
+    static pthread_once_t once = PTHREAD_ONCE_INIT;
+    static angle::TLSIndex CurrentThreadIndex = TLS_INVALID_INDEX;
+    pthread_once(&once, []() {
+      angle::qnx_terminate::Install();
+      ASSERT(CurrentThreadIndex == TLS_INVALID_INDEX);
+      CurrentThreadIndex = angle::CreateTLSIndex(nullptr);
+    });
+    return CurrentThreadIndex;
+}
+Thread *GetCurrentThreadTLS()
+{
+    angle::TLSIndex CurrentThreadIndex = GetCurrentThreadTLSIndex();
+    ASSERT(CurrentThreadIndex != TLS_INVALID_INDEX);
+    return static_cast<Thread *>(angle::GetTLSValue(CurrentThreadIndex));
+}
+void SetCurrentThreadTLS(Thread *thread)
+{
+    angle::TLSIndex CurrentThreadIndex = GetCurrentThreadTLSIndex();
+    ASSERT(CurrentThreadIndex != TLS_INVALID_INDEX);
+    angle::SetTLSValue(CurrentThreadIndex, thread);
+}
 #elif defined(ANGLE_USE_STATIC_THREAD_LOCAL_VARIABLES)
 static thread_local Thread *gCurrentThread = nullptr;
 Thread *GetCurrentThreadTLS()
@@ -140,7 +168,8 @@ void SetGlobalLastContext(gl::Context *context)
 // It also causes a flaky false positive in TSAN. http://crbug.com/1223970
 ANGLE_NO_SANITIZE_MEMORY ANGLE_NO_SANITIZE_THREAD Thread *GetCurrentThread()
 {
-#if defined(ANGLE_PLATFORM_APPLE) || defined(ANGLE_USE_STATIC_THREAD_LOCAL_VARIABLES)
+#if defined(ANGLE_PLATFORM_APPLE) || defined(ANGLE_USE_STATIC_THREAD_LOCAL_VARIABLES) || \
+    defined(__QNXNTO__)
     Thread *current = GetCurrentThreadTLS();
 #else
     Thread *current = gCurrentThread;
@@ -150,7 +179,8 @@ ANGLE_NO_SANITIZE_MEMORY ANGLE_NO_SANITIZE_THREAD Thread *GetCurrentThread()
 
 void SetContextCurrent(Thread *thread, gl::Context *context)
 {
-#if defined(ANGLE_PLATFORM_APPLE) || defined(ANGLE_USE_STATIC_THREAD_LOCAL_VARIABLES)
+#if defined(ANGLE_PLATFORM_APPLE) || defined(ANGLE_USE_STATIC_THREAD_LOCAL_VARIABLES) || \
+    defined(__QNXNTO__)
     Thread *currentThread = GetCurrentThreadTLS();
 #else
     Thread *currentThread = gCurrentThread;
diff --git src/libGLESv2/global_state.h src/libGLESv2/global_state.h
index 32c9921d9bb986e26a90172b3833b4c74b9e2a14..2faca8c99960ac94033c238d04c5f6d48f59b15e 100644
--- src/libGLESv2/global_state.h
+++ src/libGLESv2/global_state.h
@@ -21,7 +21,8 @@
 #include "libANGLE/features.h"
 #include "libANGLE/validationEGL.h"
 
-#if defined(ANGLE_PLATFORM_APPLE) || (ANGLE_PLATFORM_ANDROID) || defined(ANGLE_USE_ANDROID_TLS_SLOT)
+#if defined(ANGLE_PLATFORM_APPLE) || (ANGLE_PLATFORM_ANDROID) || defined(ANGLE_USE_ANDROID_TLS_SLOT) || \
+    defined(__QNXNTO__)
 #    include "common/tls.h"
 #endif
 
@@ -32,7 +33,8 @@ namespace egl
 class Debug;
 class Thread;
 
-#if defined(ANGLE_PLATFORM_APPLE) || defined(ANGLE_USE_STATIC_THREAD_LOCAL_VARIABLES)
+#if defined(ANGLE_PLATFORM_APPLE) || defined(ANGLE_USE_STATIC_THREAD_LOCAL_VARIABLES) || \
+    defined(__QNXNTO__)
 extern Thread *GetCurrentThreadTLS();
 extern void SetCurrentThreadTLS(Thread *thread);
 #else
@@ -126,7 +128,8 @@ namespace gl
 {
 ANGLE_INLINE Context *GetGlobalContext()
 {
-#if defined(ANGLE_PLATFORM_APPLE) || defined(ANGLE_USE_STATIC_THREAD_LOCAL_VARIABLES)
+#if defined(ANGLE_PLATFORM_APPLE) || defined(ANGLE_USE_STATIC_THREAD_LOCAL_VARIABLES) || \
+    defined(__QNXNTO__)
     egl::Thread *currentThread = egl::GetCurrentThreadTLS();
 #else
     egl::Thread *currentThread = egl::gCurrentThread;
@@ -145,7 +148,8 @@ ANGLE_INLINE Context *GetValidGlobalContext()
     }
 #endif
 
-#if defined(ANGLE_PLATFORM_APPLE) || defined(ANGLE_USE_STATIC_THREAD_LOCAL_VARIABLES)
+#if defined(ANGLE_PLATFORM_APPLE) || defined(ANGLE_USE_STATIC_THREAD_LOCAL_VARIABLES) || \
+    defined(__QNXNTO__)
     return GetCurrentValidContextTLS();
 #else
     return gCurrentValidContext;
