diff --git src/libGLESv2.gni src/libGLESv2.gni
index b7ea607dcac6618e8a197ebe2c39fced8a9264cf..293036bf616fe1fafccba27d96ca1c943fcdf942 100644
--- src/libGLESv2.gni
+++ src/libGLESv2.gni
--- src/libGLESv2.gni
+++ src/libGLESv2.gni
@@ -123,6 +123,10 @@ if (is_linux || is_qnx || is_chromeos || is_android || is_fuchsia) {
   ]
 }
 
+if (is_qnx) {
+  libangle_common_sources += [ "src/common/terminate_capture_qnx.cc" ]
+}
+
 if (is_apple) {
   libangle_common_sources += [
     "src/common/apple/ObjCPtr.h",
