diff --git components/autofill/core/browser/webdata/addresses/autofill_profile_sync_bridge.cc components/autofill/core/browser/webdata/addresses/autofill_profile_sync_bridge.cc
index 46eb03087e1e0f4b45b2a897dcda3a64beabe262..f8a1984734dc1e65fb0f408e018d8ebeda26f245 100644
--- components/autofill/core/browser/webdata/addresses/autofill_profile_sync_bridge.cc
+++ components/autofill/core/browser/webdata/addresses/autofill_profile_sync_bridge.cc
@@ -179,7 +179,8 @@ std::unique_ptr<syncer::DataBatch> AutofillProfileSyncBridge::GetDataForCommit(
     return nullptr;
   }
 
-  absl::flat_hash_set<std::string> keys_set(std::from_range, storage_keys);
+  absl::flat_hash_set<std::string> keys_set(storage_keys.begin(),
+                                           storage_keys.end());
   auto batch = std::make_unique<syncer::MutableDataBatch>();
   for (const AutofillProfile& entry : entries) {
     std::string key = GetStorageKeyFromAutofillProfile(entry);
diff --git components/autofill/core/browser/webdata/autocomplete/autocomplete_sync_bridge.cc components/autofill/core/browser/webdata/autocomplete/autocomplete_sync_bridge.cc
index ba4f58ed9338d0adfef985b2a4f2593f096e697d..b1c556e64d811e7066fd8695f0407ab652bbdcb4 100644
--- components/autofill/core/browser/webdata/autocomplete/autocomplete_sync_bridge.cc
+++ components/autofill/core/browser/webdata/autocomplete/autocomplete_sync_bridge.cc
@@ -420,7 +420,8 @@ AutocompleteSyncBridge::AutocompleteSyncBridge::GetDataForCommit(
     return nullptr;
   }
 
-  absl::flat_hash_set<std::string> keys_set(std::from_range, storage_keys);
+  absl::flat_hash_set<std::string> keys_set(storage_keys.begin(),
+                                           storage_keys.end());
   auto batch = std::make_unique<MutableDataBatch>();
   for (const AutocompleteEntry& entry : entries) {
     std::string key = GetStorageKeyFromModel(entry.key());
diff --git components/autofill/core/browser/webdata/payments/autofill_wallet_credential_sync_bridge.cc components/autofill/core/browser/webdata/payments/autofill_wallet_credential_sync_bridge.cc
index 935a805b1ad2855cdd7e01427a03d3f331a10a88..5574375b7beb047ad440908f2af9032bbe778903 100644
--- components/autofill/core/browser/webdata/payments/autofill_wallet_credential_sync_bridge.cc
+++ components/autofill/core/browser/webdata/payments/autofill_wallet_credential_sync_bridge.cc
@@ -175,8 +175,8 @@ std::unique_ptr<syncer::DataBatch>
 AutofillWalletCredentialSyncBridge::GetDataForCommit(
     StorageKeyList storage_keys) {
   DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
-  absl::flat_hash_set<std::string> storage_keys_set(std::from_range,
-                                                    storage_keys);
+  absl::flat_hash_set<std::string> storage_keys_set(storage_keys.begin(),
+                                                    storage_keys.end());
   std::vector<std::unique_ptr<ServerCvc>> filtered_server_cvc_list;
   for (std::unique_ptr<ServerCvc>& server_cvc :
        GetAutofillTable()->GetAllServerCvcs()) {
diff --git components/autofill/core/browser/webdata/payments/autofill_wallet_metadata_sync_bridge.cc components/autofill/core/browser/webdata/payments/autofill_wallet_metadata_sync_bridge.cc
index 2e44d9d042423f5a33c8c0a09e77aec49a30b260..d28db3b32223daa3e755ec760a9ef941819aab55 100644
--- components/autofill/core/browser/webdata/payments/autofill_wallet_metadata_sync_bridge.cc
+++ components/autofill/core/browser/webdata/payments/autofill_wallet_metadata_sync_bridge.cc
@@ -380,8 +380,8 @@ std::unique_ptr<syncer::DataBatch>
 AutofillWalletMetadataSyncBridge::GetDataForCommit(
     StorageKeyList storage_keys) {
   // Build a set out of the list to allow quick lookup.
-  absl::flat_hash_set<std::string> storage_keys_set(std::from_range,
-                                                    storage_keys);
+  absl::flat_hash_set<std::string> storage_keys_set(storage_keys.begin(),
+                                                    storage_keys.end());
   return GetDataImpl(std::move(storage_keys_set));
 }
 
