diff --git a/base/rand_util_posix.cc b/base/rand_util_posix.cc
index d1de8a030ffce..0d14cb47c1f1c 100644
--- a/base/rand_util_posix.cc
+++ b/base/rand_util_posix.cc
@@ -8,8 +8,10 @@
 #include <fcntl.h>
 #include <stddef.h>
 #include <stdint.h>
+#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_ANDROID)
 #include <sys/syscall.h>
 #include <sys/utsname.h>
+#endif
 #include <unistd.h>
 
 #include "base/check.h"
