diff --git third_party/blink/renderer/core/execution_context/navigator_base.cc third_party/blink/renderer/core/execution_context/navigator_base.cc
index cccfafe8f93a0228d08352c6d4f27391935848ba..2c03bce82fa975075e0ec928ec3f4db868be4ad0 100644
--- third_party/blink/renderer/core/execution_context/navigator_base.cc
+++ third_party/blink/renderer/core/execution_context/navigator_base.cc
@@ -32,7 +32,7 @@ String GetReducedNavigatorPlatform() {
   return "Win32";
 #elif BUILDFLAG(IS_FUCHSIA)
   return "";
-#elif BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
+#elif BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_QNX)
   return "Linux x86_64";
 #elif BUILDFLAG(IS_IOS)
   return "iPhone";
