diff --git chrome/browser/ui/views/frame/browser_frame_view_factory_views.cc chrome/browser/ui/views/frame/browser_frame_view_factory_views.cc
index 087a9a6ae11d93652c8fcfc59fb2c1ad0ba191d7..339b45143acc8f168bbc7ceb2b040a17929e039d 100644
--- chrome/browser/ui/views/frame/browser_frame_view_factory_views.cc
+++ chrome/browser/ui/views/frame/browser_frame_view_factory_views.cc
@@ -89,7 +89,24 @@ std::unique_ptr<BrowserFrameView> CreateBrowserFrameViewLinux(
 
   return opaque_browser_view;
 }
-#endif  // BUILDFLAG(IS_LINUX)
+#endif
+
+#if BUILDFLAG(IS_QNX)
+std::unique_ptr<BrowserFrameView> CreateBrowserFrameViewQnx(
+    BrowserWidget* widget,
+    BrowserView* browser_view) {
+  if (browser_view->browser()->is_type_picture_in_picture()) {
+    return std::make_unique<PictureInPictureBrowserFrameView>(widget,
+                                                              browser_view);
+  }
+
+  auto opaque_browser_view = std::make_unique<OpaqueBrowserFrameView>(
+      widget, browser_view, new OpaqueBrowserFrameViewLayout());
+  opaque_browser_view->InitViews();
+
+  return opaque_browser_view;
+}
+#endif  // BUILDFLAG(IS_QNX)
 
 #if BUILDFLAG(IS_WIN)
 std::unique_ptr<BrowserFrameView> CreateBrowserFrameViewWin(
@@ -119,6 +136,8 @@ std::unique_ptr<BrowserFrameView> CreateBrowserFrameView(
     BrowserView* browser_view) {
 #if BUILDFLAG(IS_WIN)
   return CreateBrowserFrameViewWin(widget, browser_view);
+#elif BUILDFLAG(IS_QNX)
+  return CreateBrowserFrameViewQnx(widget, browser_view);
 #else
   return CreateBrowserFrameViewLinux(widget, browser_view);
 #endif
