diff --git chrome/browser/supervised_user/supervised_user_navigation_observer.cc chrome/browser/supervised_user/supervised_user_navigation_observer.cc
index aa32ade62717ed0399a94942decd0d35a29db737..17937d029474167491c478ecd0a5ad227bc6d40a 100644
--- chrome/browser/supervised_user/supervised_user_navigation_observer.cc
+++ chrome/browser/supervised_user/supervised_user_navigation_observer.cc
@@ -55,12 +55,35 @@
 #include "components/supervised_user/core/browser/android/android_parental_controls.h"
 #elif BUILDFLAG(IS_CHROMEOS)
 #include "chrome/browser/supervised_user/chromeos/supervised_user_web_content_handler_impl.h"
+#elif BUILDFLAG(IS_QNX)
+#include "chrome/browser/supervised_user/chrome_supervised_user_web_content_handler_base.h"
 #elif BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_WIN)
 #include "chrome/browser/supervised_user/linux_mac_windows/supervised_user_web_content_handler_impl.h"
 #endif
 
 namespace {
 
+#if BUILDFLAG(IS_QNX)
+class QnxSupervisedUserWebContentHandler final
+    : public ChromeSupervisedUserWebContentHandlerBase {
+ public:
+  QnxSupervisedUserWebContentHandler(content::WebContents* web_contents,
+                                     content::FrameTreeNodeId frame_id,
+                                     int64_t navigation_id)
+      : ChromeSupervisedUserWebContentHandlerBase(web_contents,
+                                                  frame_id,
+                                                  navigation_id) {}
+
+  void RequestLocalApproval(
+      const GURL& target_url,
+      supervised_user::WebFilteringResult filtering_result,
+      const std::u16string& child_display_name,
+      ApprovalRequestInitiatedCallback callback) override {
+    std::move(callback).Run(false);
+  }
+};
+#endif
+
 std::unique_ptr<supervised_user::WebContentHandler> CreateWebContentHandler(
     content::WebContents* web_contents,
     GURL url,
@@ -75,6 +98,9 @@ std::unique_ptr<supervised_user::WebContentHandler> CreateWebContentHandler(
 #elif BUILDFLAG(IS_ANDROID)
   return std::make_unique<SupervisedUserWebContentHandlerImpl>(
       web_contents, frame_id, navigation_id);
+#elif BUILDFLAG(IS_QNX)
+  return std::make_unique<QnxSupervisedUserWebContentHandler>(
+      web_contents, frame_id, navigation_id);
 #elif BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_WIN)
   return std::make_unique<SupervisedUserWebContentHandlerImpl>(
       web_contents, frame_id, navigation_id);
