diff --git a/chrome/browser/ui/find_bar/BUILD.gn b/chrome/browser/ui/find_bar/BUILD.gn
index 1775f6db1bac2..5dcdab7f67898 100644
--- a/chrome/browser/ui/find_bar/BUILD.gn
+++ b/chrome/browser/ui/find_bar/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//build/config/chromeos/args.gni")
 
-assert(is_win || is_mac || is_linux || is_chromeos || is_android)
+assert(is_win || is_mac || is_linux || is_chromeos || is_android || is_qnx)
 
 source_set("find_bar") {
   sources = [
