diff --git chrome/browser/ui/autofill/BUILD.gn chrome/browser/ui/autofill/BUILD.gn
index e59abb96c6dcd..14fa1672e5db0 100644
--- chrome/browser/ui/autofill/BUILD.gn
+++ chrome/browser/ui/autofill/BUILD.gn
@@ -12,7 +12,7 @@ if (is_android) {
   import("//build/config/android/rules.gni")
 }
 
-assert(is_win || is_mac || is_linux || is_chromeos || is_android)
+assert(is_win || is_mac || is_linux || is_chromeos || is_android || is_qnx)
 
 if (is_android) {
   java_cpp_enum("autofill_ai_ui_java_enums") {
diff --git chrome/browser/ui/autofill/payments/BUILD.gn chrome/browser/ui/autofill/payments/BUILD.gn
index b5bb39ca408ba..7384ee11d37a8 100644
--- chrome/browser/ui/autofill/payments/BUILD.gn
+++ chrome/browser/ui/autofill/payments/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//build/config/ui.gni")
 
-assert(is_win || is_mac || is_linux || is_chromeos || is_android)
+assert(is_win || is_mac || is_linux || is_chromeos || is_android || is_qnx)
 
 source_set("payments") {
   sources = [
diff --git chrome/browser/ui/blocked_content/BUILD.gn chrome/browser/ui/blocked_content/BUILD.gn
index 34d269d0d6abd..1f8c42003fa57 100644
--- chrome/browser/ui/blocked_content/BUILD.gn
+++ chrome/browser/ui/blocked_content/BUILD.gn
@@ -5,7 +5,7 @@
 import("//build/config/chromeos/args.gni")
 import("//build/config/ozone.gni")
 
-assert(is_win || is_mac || is_linux || is_chromeos || is_android)
+assert(is_win || is_mac || is_linux || is_chromeos || is_android || is_qnx)
 
 source_set("blocked_content") {
   sources = [
diff --git chrome/browser/ui/bluetooth/BUILD.gn chrome/browser/ui/bluetooth/BUILD.gn
index ba0dd9c8a3f9f..71ee147bf4717 100644
--- chrome/browser/ui/bluetooth/BUILD.gn
+++ chrome/browser/ui/bluetooth/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos || is_android)
+assert(is_win || is_mac || is_linux || is_chromeos || is_android || is_qnx)
 
 source_set("bluetooth") {
   sources = [ "bluetooth_dialogs.h" ]
diff --git chrome/browser/ui/bookmarks/BUILD.gn chrome/browser/ui/bookmarks/BUILD.gn
index bef3d6ee87754..a3988cc3ff833 100644
--- chrome/browser/ui/bookmarks/BUILD.gn
+++ chrome/browser/ui/bookmarks/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 source_set("bookmarks") {
   sources = [
diff --git chrome/browser/ui/commerce/BUILD.gn chrome/browser/ui/commerce/BUILD.gn
index 74816852d07cd..03c1ed1b18206 100644
--- chrome/browser/ui/commerce/BUILD.gn
+++ chrome/browser/ui/commerce/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 source_set("commerce") {
   sources = [
diff --git chrome/browser/ui/content_settings/BUILD.gn chrome/browser/ui/content_settings/BUILD.gn
index bcea081489749..ebf7566177da2 100644
--- chrome/browser/ui/content_settings/BUILD.gn
+++ chrome/browser/ui/content_settings/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 source_set("content_settings") {
   sources = []
diff --git chrome/browser/ui/contextual_search/BUILD.gn chrome/browser/ui/contextual_search/BUILD.gn
index e4a31afded7c4..dced6198c9635 100644
--- chrome/browser/ui/contextual_search/BUILD.gn
+++ chrome/browser/ui/contextual_search/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//pdf/features.gni")
 
-assert(is_win || is_mac || is_linux || is_chromeos || is_android)
+assert(is_win || is_mac || is_linux || is_chromeos || is_android || is_qnx)
 
 source_set("contextual_search") {
   sources = [
diff --git chrome/browser/ui/cookie_controls/BUILD.gn chrome/browser/ui/cookie_controls/BUILD.gn
index c2606478d499d..1ae303956af0a 100644
--- chrome/browser/ui/cookie_controls/BUILD.gn
+++ chrome/browser/ui/cookie_controls/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos || is_android)
+assert(is_win || is_mac || is_linux || is_chromeos || is_android || is_qnx)
 
 source_set("cookie_controls") {
   sources = []
diff --git chrome/browser/ui/customize_chrome/BUILD.gn chrome/browser/ui/customize_chrome/BUILD.gn
index 85904e44f6847..f16f77ba61073 100644
--- chrome/browser/ui/customize_chrome/BUILD.gn
+++ chrome/browser/ui/customize_chrome/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 source_set("customize_chrome") {
   sources = [ "side_panel_controller.h" ]
diff --git chrome/browser/ui/digital_credentials/BUILD.gn chrome/browser/ui/digital_credentials/BUILD.gn
index 8295df15d6cd7..4beba7a2be1b9 100644
--- chrome/browser/ui/digital_credentials/BUILD.gn
+++ chrome/browser/ui/digital_credentials/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos || is_android)
+assert(is_win || is_mac || is_linux || is_chromeos || is_android || is_qnx)
 
 source_set("digital_credentials") {
   sources = []
diff --git chrome/browser/ui/download/BUILD.gn chrome/browser/ui/download/BUILD.gn
index 10932f1075a48..2156777643ae0 100644
--- chrome/browser/ui/download/BUILD.gn
+++ chrome/browser/ui/download/BUILD.gn
@@ -3,7 +3,7 @@
 # found in the LICENSE file.
 
 # Download bubble UI is not used on ChromeOS.
-assert(is_win || is_mac || is_linux)
+assert(is_win || is_mac || is_linux || is_qnx)
 
 source_set("download") {
   sources = [
diff --git chrome/browser/ui/exclusive_access/BUILD.gn chrome/browser/ui/exclusive_access/BUILD.gn
index 72691dc8f6b03..3c11d7ef2fdd7 100644
--- chrome/browser/ui/exclusive_access/BUILD.gn
+++ chrome/browser/ui/exclusive_access/BUILD.gn
@@ -6,7 +6,7 @@ import("//build/config/chromeos/args.gni")
 import("//build/config/ozone.gni")
 import("//extensions/buildflags/buildflags.gni")
 
-assert(is_win || is_mac || is_linux || is_chromeos || is_android)
+assert(is_win || is_mac || is_linux || is_chromeos || is_android || is_qnx)
 
 source_set("exclusive_access") {
   sources = [
diff --git chrome/browser/ui/file_system_access/BUILD.gn chrome/browser/ui/file_system_access/BUILD.gn
index d156143edb301..5131e58284fae 100644
--- chrome/browser/ui/file_system_access/BUILD.gn
+++ chrome/browser/ui/file_system_access/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//components/safe_browsing/buildflags.gni")
 
-assert(is_win || is_mac || is_linux || is_chromeos || is_android)
+assert(is_win || is_mac || is_linux || is_chromeos || is_android || is_qnx)
 
 source_set("file_system_access") {
   sources = [
diff --git chrome/browser/ui/frame/BUILD.gn chrome/browser/ui/frame/BUILD.gn
index 76c081e40f831..d8fb1271d1e9e 100644
--- chrome/browser/ui/frame/BUILD.gn
+++ chrome/browser/ui/frame/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 source_set("frame") {
   sources = [
diff --git chrome/browser/ui/global_error/BUILD.gn chrome/browser/ui/global_error/BUILD.gn
index cf8504a0688ec..616ca337eb745 100644
--- chrome/browser/ui/global_error/BUILD.gn
+++ chrome/browser/ui/global_error/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 source_set("global_error") {
   sources = [
diff --git chrome/browser/ui/global_media_controls/BUILD.gn chrome/browser/ui/global_media_controls/BUILD.gn
index 63840141f5623..4dcf6fbab86a3 100644
--- chrome/browser/ui/global_media_controls/BUILD.gn
+++ chrome/browser/ui/global_media_controls/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//chrome/common/features.gni")
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 source_set("global_media_controls") {
   sources = [
diff --git chrome/browser/ui/hats/BUILD.gn chrome/browser/ui/hats/BUILD.gn
index af8c173f8e17f..fd34a4bba845d 100644
--- chrome/browser/ui/hats/BUILD.gn
+++ chrome/browser/ui/hats/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//pdf/features.gni")
 
-assert(is_win || is_mac || is_linux || is_chromeos || is_android)
+assert(is_win || is_mac || is_linux || is_chromeos || is_android || is_qnx)
 
 source_set("hats") {
   sources = [
diff --git chrome/browser/ui/hid/BUILD.gn chrome/browser/ui/hid/BUILD.gn
index c1b7d6709724c..cdae101f3f0ed 100644
--- chrome/browser/ui/hid/BUILD.gn
+++ chrome/browser/ui/hid/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 source_set("hid") {
   sources = [
diff --git chrome/browser/ui/hung_renderer/BUILD.gn chrome/browser/ui/hung_renderer/BUILD.gn
index 09043b4d8e207..793ca1a871f66 100644
--- chrome/browser/ui/hung_renderer/BUILD.gn
+++ chrome/browser/ui/hung_renderer/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//build/config/chromeos/args.gni")
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 source_set("hung_renderer") {
   sources = [
diff --git chrome/browser/ui/interventions/BUILD.gn chrome/browser/ui/interventions/BUILD.gn
index 379c4ef148280..396b69bb23371 100644
--- chrome/browser/ui/interventions/BUILD.gn
+++ chrome/browser/ui/interventions/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos || is_android)
+assert(is_win || is_mac || is_linux || is_chromeos || is_android || is_qnx)
 
 source_set("interventions") {
   sources = [
diff --git chrome/browser/ui/javascript_dialogs/BUILD.gn chrome/browser/ui/javascript_dialogs/BUILD.gn
index f824962b98548..714ed6ee8c301 100644
--- chrome/browser/ui/javascript_dialogs/BUILD.gn
+++ chrome/browser/ui/javascript_dialogs/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos || is_android)
+assert(is_win || is_mac || is_linux || is_chromeos || is_android || is_qnx)
 
 source_set("javascript_dialogs") {
   sources = [
diff --git chrome/browser/ui/login/BUILD.gn chrome/browser/ui/login/BUILD.gn
index 6d6c7cab0dec1..d2671f9afb193 100644
--- chrome/browser/ui/login/BUILD.gn
+++ chrome/browser/ui/login/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos || is_android)
+assert(is_win || is_mac || is_linux || is_chromeos || is_android || is_qnx)
 
 source_set("login") {
   sources = [
diff --git chrome/browser/ui/media_router/BUILD.gn chrome/browser/ui/media_router/BUILD.gn
index 6213b1cfd517b..c4497e2347159 100644
--- chrome/browser/ui/media_router/BUILD.gn
+++ chrome/browser/ui/media_router/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 source_set("media_router") {
   sources = [
diff --git chrome/browser/ui/omnibox/BUILD.gn chrome/browser/ui/omnibox/BUILD.gn
index 1cd3297187e0b..b9e84e6fdd774 100644
--- chrome/browser/ui/omnibox/BUILD.gn
+++ chrome/browser/ui/omnibox/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//build/config/chromeos/args.gni")
 
-assert(is_win || is_mac || is_linux || is_chromeos || is_android)
+assert(is_win || is_mac || is_linux || is_chromeos || is_android || is_qnx)
 
 # The public interface has no circular dependencies.
 source_set("omnibox") {
diff --git chrome/browser/ui/page_action/BUILD.gn chrome/browser/ui/page_action/BUILD.gn
index 44d8e922eacf7..25249f4e98206 100644
--- chrome/browser/ui/page_action/BUILD.gn
+++ chrome/browser/ui/page_action/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos || is_android)
+assert(is_win || is_mac || is_linux || is_chromeos || is_android || is_qnx)
 
 source_set("icon_type") {
   sources = [
diff --git chrome/browser/ui/page_info/BUILD.gn chrome/browser/ui/page_info/BUILD.gn
index a33b10bde2ade..ddee51bfb2420 100644
--- chrome/browser/ui/page_info/BUILD.gn
+++ chrome/browser/ui/page_info/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos || is_android)
+assert(is_win || is_mac || is_linux || is_chromeos || is_android || is_qnx)
 
 source_set("page_info") {
   sources = [
diff --git chrome/browser/ui/permission_bubble/BUILD.gn chrome/browser/ui/permission_bubble/BUILD.gn
index f5a7b55d91779..ef54bf3b9137c 100644
--- chrome/browser/ui/permission_bubble/BUILD.gn
+++ chrome/browser/ui/permission_bubble/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 source_set("permission_bubble") {
   sources = [ "permission_prompt.h" ]
diff --git chrome/browser/ui/prefs/BUILD.gn chrome/browser/ui/prefs/BUILD.gn
index 955e92882c1c0..2373f8455221c 100644
--- chrome/browser/ui/prefs/BUILD.gn
+++ chrome/browser/ui/prefs/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//extensions/buildflags/buildflags.gni")
 
-assert(is_win || is_mac || is_linux || is_chromeos || is_android)
+assert(is_win || is_mac || is_linux || is_chromeos || is_android || is_qnx)
 
 source_set("prefs") {
   sources = [
diff --git chrome/browser/ui/qrcode_generator/BUILD.gn chrome/browser/ui/qrcode_generator/BUILD.gn
index 36f97343c5230..de199d30308bb 100644
--- chrome/browser/ui/qrcode_generator/BUILD.gn
+++ chrome/browser/ui/qrcode_generator/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//build/config/ui.gni")
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 assert(toolkit_views)
 
 source_set("qrcode_generator") {
diff --git chrome/browser/ui/read_anything/BUILD.gn chrome/browser/ui/read_anything/BUILD.gn
index b9069e63fa987..cde76bacea84c 100644
--- chrome/browser/ui/read_anything/BUILD.gn
+++ chrome/browser/ui/read_anything/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 import("//build/config/compiler/compiler.gni")
 import("//chrome/common/features.gni")
diff --git chrome/browser/ui/safety_hub/BUILD.gn chrome/browser/ui/safety_hub/BUILD.gn
index 3f28fedbba0df..ee94a89c47f87 100644
--- chrome/browser/ui/safety_hub/BUILD.gn
+++ chrome/browser/ui/safety_hub/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//extensions/buildflags/buildflags.gni")
 
-assert(is_win || is_mac || is_linux || is_chromeos || is_android)
+assert(is_win || is_mac || is_linux || is_chromeos || is_android || is_qnx)
 
 source_set("safety_hub") {
   sources = [
diff --git chrome/browser/ui/search/BUILD.gn chrome/browser/ui/search/BUILD.gn
index c693a55367862..e88ed145a56a0 100644
--- chrome/browser/ui/search/BUILD.gn
+++ chrome/browser/ui/search/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//build/config/chromeos/args.gni")
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 source_set("search") {
   sources = [
diff --git chrome/browser/ui/send_tab_to_self/BUILD.gn chrome/browser/ui/send_tab_to_self/BUILD.gn
index 8354e1b553d4c..7269973e1018f 100644
--- chrome/browser/ui/send_tab_to_self/BUILD.gn
+++ chrome/browser/ui/send_tab_to_self/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//build/config/ui.gni")
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 assert(toolkit_views)
 
 source_set("send_tab_to_self") {
diff --git chrome/browser/ui/shared_highlighting/BUILD.gn chrome/browser/ui/shared_highlighting/BUILD.gn
index e82971c2c7496..ae22cf4ec81b1 100644
--- chrome/browser/ui/shared_highlighting/BUILD.gn
+++ chrome/browser/ui/shared_highlighting/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 source_set("shared_highlighting") {
   sources = [
diff --git chrome/browser/ui/sharing_hub/BUILD.gn chrome/browser/ui/sharing_hub/BUILD.gn
index 1086f6a8735f2..135667ddbb33c 100644
--- chrome/browser/ui/sharing_hub/BUILD.gn
+++ chrome/browser/ui/sharing_hub/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//build/config/ui.gni")
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 assert(toolkit_views)
 
 source_set("sharing_hub") {
diff --git chrome/browser/ui/shortcuts/BUILD.gn chrome/browser/ui/shortcuts/BUILD.gn
index f7d200ff40860..9cf08e6dacc99 100644
--- chrome/browser/ui/shortcuts/BUILD.gn
+++ chrome/browser/ui/shortcuts/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux)
+assert(is_win || is_mac || is_linux || is_qnx)
 
 source_set("shortcuts") {
   sources = [
diff --git chrome/browser/ui/startup/default_browser_prompt/BUILD.gn chrome/browser/ui/startup/default_browser_prompt/BUILD.gn
index ab847cc12810c..4d72b9b903a55 100644
--- chrome/browser/ui/startup/default_browser_prompt/BUILD.gn
+++ chrome/browser/ui/startup/default_browser_prompt/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux,
+assert(is_win || is_mac || is_linux || is_qnx,
        "Default browser prompts exist only on non-Chromeos desktop platforms")
 
 source_set("default_browser_prompt") {
diff --git chrome/browser/ui/tab_contents/BUILD.gn chrome/browser/ui/tab_contents/BUILD.gn
index ae360a3889187..65c76fde1cc5f 100644
--- chrome/browser/ui/tab_contents/BUILD.gn
+++ chrome/browser/ui/tab_contents/BUILD.gn
@@ -5,7 +5,7 @@
 import("//components/enterprise/buildflags/buildflags.gni")
 import("//extensions/buildflags/buildflags.gni")
 
-assert(is_win || is_mac || is_linux || is_chromeos || is_android)
+assert(is_win || is_mac || is_linux || is_chromeos || is_android || is_qnx)
 
 source_set("tab_contents") {
   sources = [
diff --git chrome/browser/ui/tabs/BUILD.gn chrome/browser/ui/tabs/BUILD.gn
index 823f383fe2ade..ad6f3a5fd6e74 100644
--- chrome/browser/ui/tabs/BUILD.gn
+++ chrome/browser/ui/tabs/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos || is_android)
+assert(is_win || is_mac || is_linux || is_chromeos || is_android || is_qnx)
 
 import("//chrome/common/features.gni")
 
diff --git chrome/browser/ui/tabs/alert/BUILD.gn chrome/browser/ui/tabs/alert/BUILD.gn
index 6af0314761dcf..8e71bddc82560 100644
--- chrome/browser/ui/tabs/alert/BUILD.gn
+++ chrome/browser/ui/tabs/alert/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos || is_android)
+assert(is_win || is_mac || is_linux || is_chromeos || is_android || is_qnx)
 
 import("//chrome/common/features.gni")
 
diff --git chrome/browser/ui/tabs/saved_tab_groups/BUILD.gn chrome/browser/ui/tabs/saved_tab_groups/BUILD.gn
index 3034fae1f7132..283dbb23b47bc 100644
--- chrome/browser/ui/tabs/saved_tab_groups/BUILD.gn
+++ chrome/browser/ui/tabs/saved_tab_groups/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 import("//chrome/common/features.gni")
 
diff --git chrome/browser/ui/tabs/tab_strip_api/BUILD.gn chrome/browser/ui/tabs/tab_strip_api/BUILD.gn
index 7224d1e47d55d..f105cbcfa5446 100644
--- chrome/browser/ui/tabs/tab_strip_api/BUILD.gn
+++ chrome/browser/ui/tabs/tab_strip_api/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 source_set("tab_strip_api") {
   sources = [
diff --git chrome/browser/ui/toasts/BUILD.gn chrome/browser/ui/toasts/BUILD.gn
index 53ee87d01b9b4..3c917ccdf97bf 100644
--- chrome/browser/ui/toasts/BUILD.gn
+++ chrome/browser/ui/toasts/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 import("//build/config/chrome_build.gni")
 import("//build/config/ui.gni")
diff --git chrome/browser/ui/toolbar/cast/BUILD.gn chrome/browser/ui/toolbar/cast/BUILD.gn
index 1a16ef287f00f..7cbd8faa460fb 100644
--- chrome/browser/ui/toolbar/cast/BUILD.gn
+++ chrome/browser/ui/toolbar/cast/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 source_set("cast") {
   sources = [
diff --git chrome/browser/ui/user_education/BUILD.gn chrome/browser/ui/user_education/BUILD.gn
index 53db4a588d5b1..b24c762a140a8 100644
--- chrome/browser/ui/user_education/BUILD.gn
+++ chrome/browser/ui/user_education/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//build/config/chromeos/args.gni")
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 # This is the public interface for the user education service.
 source_set("user_education") {
diff --git chrome/browser/ui/views/bubble/BUILD.gn chrome/browser/ui/views/bubble/BUILD.gn
index 79de1873b913e..da74f455e5892 100644
--- chrome/browser/ui/views/bubble/BUILD.gn
+++ chrome/browser/ui/views/bubble/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 import("//build/config/ui.gni")
 
diff --git chrome/browser/ui/views/contextual_tasks/BUILD.gn chrome/browser/ui/views/contextual_tasks/BUILD.gn
index 413319ac0f647..aa0dfea1ebd2e 100644
--- chrome/browser/ui/views/contextual_tasks/BUILD.gn
+++ chrome/browser/ui/views/contextual_tasks/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD - style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 source_set("contextual_tasks") {
   sources = [
diff --git chrome/browser/ui/views/download/BUILD.gn chrome/browser/ui/views/download/BUILD.gn
index 1863fa68101ed..55fadce9cf45c 100644
--- chrome/browser/ui/views/download/BUILD.gn
+++ chrome/browser/ui/views/download/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD - style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 source_set("download") {
   sources = [
diff --git chrome/browser/ui/views/location_bar/BUILD.gn chrome/browser/ui/views/location_bar/BUILD.gn
index c524ea2cffdcd..f422cfae40c9a 100644
--- chrome/browser/ui/views/location_bar/BUILD.gn
+++ chrome/browser/ui/views/location_bar/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 import("//build/config/ui.gni")
 import("//chrome/common/features.gni")
diff --git chrome/browser/ui/views/location_bar/cookie_controls/BUILD.gn chrome/browser/ui/views/location_bar/cookie_controls/BUILD.gn
index f584ea506034a..6327634779785 100644
--- chrome/browser/ui/views/location_bar/cookie_controls/BUILD.gn
+++ chrome/browser/ui/views/location_bar/cookie_controls/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//build/config/ui.gni")
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 source_set("cookie_controls") {
   sources = []
diff --git chrome/browser/ui/views/send_tab_to_self/BUILD.gn chrome/browser/ui/views/send_tab_to_self/BUILD.gn
index c6d688ae05d53..7fe1424828592 100644
--- chrome/browser/ui/views/send_tab_to_self/BUILD.gn
+++ chrome/browser/ui/views/send_tab_to_self/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 source_set("send_tab_to_self") {
   public = [
diff --git chrome/browser/ui/views/side_panel/BUILD.gn chrome/browser/ui/views/side_panel/BUILD.gn
index e3bc99078f473..dbe5f49e08052 100644
--- chrome/browser/ui/views/side_panel/BUILD.gn
+++ chrome/browser/ui/views/side_panel/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD - style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 import("//chrome/common/features.gni")
 import("//extensions/buildflags/buildflags.gni")
diff --git chrome/browser/ui/webui/accessibility/BUILD.gn chrome/browser/ui/webui/accessibility/BUILD.gn
index 8ef85d23c153f..015ecbb20a649 100644
--- chrome/browser/ui/webui/accessibility/BUILD.gn
+++ chrome/browser/ui/webui/accessibility/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos || is_android)
+assert(is_win || is_mac || is_linux || is_chromeos || is_android || is_qnx)
 
 source_set("accessibility") {
   sources = [ "accessibility_ui.h" ]
diff --git chrome/browser/ui/webui/app_home/BUILD.gn chrome/browser/ui/webui/app_home/BUILD.gn
index 3ce7cddfa774b..0d81f7fe9377e 100644
--- chrome/browser/ui/webui/app_home/BUILD.gn
+++ chrome/browser/ui/webui/app_home/BUILD.gn
@@ -5,7 +5,7 @@
 import("//extensions/buildflags/buildflags.gni")
 import("//mojo/public/tools/bindings/mojom.gni")
 
-assert(is_win || is_mac || is_linux)
+assert(is_win || is_mac || is_linux || is_qnx)
 
 mojom("mojo_bindings") {
   sources = [ "app_home.mojom" ]
diff --git chrome/browser/ui/webui/app_management/BUILD.gn chrome/browser/ui/webui/app_management/BUILD.gn
index b21f22cd6b448..0783240eb585a 100644
--- chrome/browser/ui/webui/app_management/BUILD.gn
+++ chrome/browser/ui/webui/app_management/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 source_set("app_management") {
   sources = [
diff --git chrome/browser/ui/webui/app_settings/BUILD.gn chrome/browser/ui/webui/app_settings/BUILD.gn
index 54f42ea5e5bae..278db16d7cabf 100644
--- chrome/browser/ui/webui/app_settings/BUILD.gn
+++ chrome/browser/ui/webui/app_settings/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux)
+assert(is_win || is_mac || is_linux || is_qnx)
 
 source_set("app_settings") {
   sources = [
diff --git chrome/browser/ui/webui/cr_components/composebox/BUILD.gn chrome/browser/ui/webui/cr_components/composebox/BUILD.gn
index 7265f2390d9fb..39a83471b7d5c 100644
--- chrome/browser/ui/webui/cr_components/composebox/BUILD.gn
+++ chrome/browser/ui/webui/cr_components/composebox/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 source_set("composebox") {
   sources = [
diff --git chrome/browser/ui/webui/cr_components/searchbox/BUILD.gn chrome/browser/ui/webui/cr_components/searchbox/BUILD.gn
index be84e7baff74e..c016102fca9e6 100644
--- chrome/browser/ui/webui/cr_components/searchbox/BUILD.gn
+++ chrome/browser/ui/webui/cr_components/searchbox/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 source_set("searchbox") {
   sources = [
diff --git chrome/browser/ui/webui/safe_browsing/BUILD.gn chrome/browser/ui/webui/safe_browsing/BUILD.gn
index dca19f26e0b41..1feaa73119b9e 100644
--- chrome/browser/ui/webui/safe_browsing/BUILD.gn
+++ chrome/browser/ui/webui/safe_browsing/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos || is_android)
+assert(is_win || is_mac || is_linux || is_chromeos || is_android || is_qnx)
 
 source_set("safe_browsing") {
   sources = [
diff --git chrome/browser/ui/webui/searchbox/BUILD.gn chrome/browser/ui/webui/searchbox/BUILD.gn
index 1f00f2e01c7ec..4e89caf86879c 100644
--- chrome/browser/ui/webui/searchbox/BUILD.gn
+++ chrome/browser/ui/webui/searchbox/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 source_set("searchbox") {
   sources = [
diff --git chrome/browser/ui/webui/settings/BUILD.gn chrome/browser/ui/webui/settings/BUILD.gn
index 8402563c745b3..96e42a1db98c4 100644
--- chrome/browser/ui/webui/settings/BUILD.gn
+++ chrome/browser/ui/webui/settings/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//components/signin/features.gni")
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 # TODO(crbug.com/369369575): complete modularization of this component.
 # At the moment, there are a lot of dependencies from headers to //c/b and
diff --git chrome/browser/ui/webui/signin/history_sync_optin/BUILD.gn chrome/browser/ui/webui/signin/history_sync_optin/BUILD.gn
index 86cb680353af4..7181c4607c1ec 100644
--- chrome/browser/ui/webui/signin/history_sync_optin/BUILD.gn
+++ chrome/browser/ui/webui/signin/history_sync_optin/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//mojo/public/tools/bindings/mojom.gni")
 
-assert(is_win || is_mac || is_linux)
+assert(is_win || is_mac || is_linux || is_qnx)
 
 mojom("mojo_bindings") {
   sources = [ "history_sync_optin.mojom" ]
diff --git chrome/browser/ui/webui/tab_search/BUILD.gn chrome/browser/ui/webui/tab_search/BUILD.gn
index 48539b48e5b58..9fd2b694d12d5 100644
--- chrome/browser/ui/webui/tab_search/BUILD.gn
+++ chrome/browser/ui/webui/tab_search/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//mojo/public/tools/bindings/mojom.gni")
 
-assert(is_linux || is_chromeos || is_win || is_mac)
+assert(is_linux || is_chromeos || is_win || is_mac || is_qnx)
 assert(!is_fuchsia, "Fuchsia shouldn't use anything in //chrome")
 
 mojom("mojo_bindings") {
diff --git chrome/browser/ui/webui/updater/BUILD.gn chrome/browser/ui/webui/updater/BUILD.gn
index ceede34a2e559..614f3ac054582 100644
--- chrome/browser/ui/webui/updater/BUILD.gn
+++ chrome/browser/ui/webui/updater/BUILD.gn
@@ -4,7 +4,7 @@
 
 import("//mojo/public/tools/bindings/mojom.gni")
 
-assert(is_mac || is_win || is_linux)
+assert(is_mac || is_win || is_linux || is_qnx)
 
 mojom("mojo_bindings") {
   sources = [ "updater_ui.mojom" ]
diff --git chrome/browser/ui/webui/util/BUILD.gn chrome/browser/ui/webui/util/BUILD.gn
index fec7b4da30017..55a49017ce3ff 100644
--- chrome/browser/ui/webui/util/BUILD.gn
+++ chrome/browser/ui/webui/util/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos)
+assert(is_win || is_mac || is_linux || is_chromeos || is_qnx)
 
 source_set("util") {
   public = [ "image_util.h" ]
diff --git chrome/browser/ui/zoom/BUILD.gn chrome/browser/ui/zoom/BUILD.gn
index 302e928629ee5..06e76d9fe2da2 100644
--- chrome/browser/ui/zoom/BUILD.gn
+++ chrome/browser/ui/zoom/BUILD.gn
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-assert(is_win || is_mac || is_linux || is_chromeos || is_android)
+assert(is_win || is_mac || is_linux || is_chromeos || is_android || is_qnx)
 
 source_set("zoom") {
   sources = [
