diff --git chrome/browser/BUILD.gn chrome/browser/BUILD.gn
--- chrome/browser/BUILD.gn
+++ chrome/browser/BUILD.gn
@@ -2501,6 +2501,12 @@
   ]
 
   if (is_qnx) {
+    deps += [
+      "//chrome/browser/policy:battery_saver_qnx",
+    ]
+  }
+
+  if (is_qnx) {
     sources += [ "platform_util_qnx.cc" ]
   }
 
@@ -7034,6 +7040,13 @@
   if (is_qnx) {
     sources += [
       "first_run/upgrade_util_qnx.cc",
+      "notifications/notification_platform_bridge_linux_qnx.cc",
+    ]
+  }
+
+  if (is_qnx && !is_linux) {
+    sources += [
+      "memory_details_linux.cc",
     ]
   }
 

