diff --git chrome/browser/ui/BUILD.gn chrome/browser/ui/BUILD.gn
index 00b9254990f24..3d927ee1e4e34 100644
--- chrome/browser/ui/BUILD.gn
+++ chrome/browser/ui/BUILD.gn
@@ -2925,7 +2925,7 @@ static_library("ui") {
     }
   }

-  if (is_win || is_mac || is_linux || is_chromeos) {
+  if (is_win || is_mac || is_linux || is_chromeos || is_qnx) {
     sources += [
       "frame/window_frame_util.cc",
       "frame/window_frame_util.h",
@@ -2978,6 +2978,7 @@ static_library("ui") {
       "views/profiles/signin_view_controller_delegate_views.h",
     ]
     deps += [
+      "//chrome/browser/ui/webui/signin/history_sync_optin:mojo_bindings",
       "//chrome/browser/contextual_tasks:ui",
       "//chrome/browser/lifetime:termination_notification",
       "//chrome/browser/search/background",
