diff --git chrome/utility/BUILD.gn chrome/utility/BUILD.gn
index 09a55515091dcf8acdb23023aad8d6eb8d16de3c..9bb0eb193906ccd22bed9876a7ddd5bcc6cf51f5 100644
--- chrome/utility/BUILD.gn
+++ chrome/utility/BUILD.gn
@@ -102,7 +102,7 @@ static_library("utility") {
       "importer/profile_import_impl.cc",
       "importer/profile_import_impl.h",
     ]
-    if (!is_chromeos) {
+    if (!is_chromeos && !is_qnx) {
       sources += [
         "importer/firefox_importer.cc",
         "importer/firefox_importer.h",
@@ -114,9 +114,9 @@ static_library("utility") {
         "importer/safari_importer.mm",
       ]
     }
-    if (!is_mac && !is_chromeos) {
-      # NSS decryptor is not needed on ChromeOS. Additionally, it depends on
-      # system NSS that is unavailable on MacOS/Fuchsia.
+    if (!is_mac && !is_chromeos && !is_qnx) {
+      # NSS decryptor is not needed on ChromeOS or QNX. Additionally, it depends
+      # on system NSS that is unavailable on MacOS/Fuchsia.
       sources += [
         "importer/nss_decryptor.cc",
         "importer/nss_decryptor.h",
