diff --git chrome/browser/ui/views/frame/browser_view.cc chrome/browser/ui/views/frame/browser_view.cc
--- chrome/browser/ui/views/frame/browser_view.cc
+++ chrome/browser/ui/views/frame/browser_view.cc
@@ -5988,10 +5988,12 @@ void BrowserView::ShowHatsDialog(
     base::OnceClosure success_callback,
     base::OnceClosure failure_callback,
     const SurveyBitsData& product_specific_bits_data,
     const SurveyStringData& product_specific_string_data) {
+#if !BUILDFLAG(IS_QNX)
   // Self deleting on close.
   new HatsNextWebDialog(browser(), site_id, hats_histogram_name,
                         hats_survey_ukm_id, std::move(success_callback),
                         std::move(failure_callback), product_specific_bits_data,
                         product_specific_string_data);
+#endif
 }
