diff --git chrome/browser/ui/passwords/BUILD.gn chrome/browser/ui/passwords/BUILD.gn
index 33e975c663321..f63d29f72f356 100644
--- chrome/browser/ui/passwords/BUILD.gn
+++ chrome/browser/ui/passwords/BUILD.gn
@@ -53,7 +53,7 @@ source_set("passwords") {
     deps += [ "//chrome/browser/ui/toasts/api:toasts" ]
   }
 
-  if (is_win || is_mac || is_linux || is_chromeos) {
+  if (is_win || is_mac || is_linux || is_chromeos || is_qnx) {
     sources += [
       "password_cross_domain_confirmation_popup_controller_impl.h",
       "password_cross_domain_confirmation_popup_controller_interface.h",
@@ -155,7 +155,7 @@ source_set("impl") {
     ]
   }
 
-  if (is_win || is_mac || is_linux || is_chromeos) {
+  if (is_win || is_mac || is_linux || is_chromeos || is_qnx) {
     sources += [ "password_cross_domain_confirmation_popup_controller_impl.cc" ]
   }
 
