diff --git chrome/browser/ui/webui/chrome_web_ui_controller_factory.cc chrome/browser/ui/webui/chrome_web_ui_controller_factory.cc
index 64348336c6b24fb0ed59d6290690a2704248b993..13427cf237ef19e146e0d70a7f82c6480fb909bc 100644
--- chrome/browser/ui/webui/chrome_web_ui_controller_factory.cc
+++ chrome/browser/ui/webui/chrome_web_ui_controller_factory.cc
@@ -171,7 +171,8 @@ WebUIFactoryFunction GetWebUIFactoryFunction(Profile* profile,
 #endif  // BUILDFLAG(ENABLE_DEVTOOLS_FRONTEND)
 }
 
-#if !BUILDFLAG(IS_ANDROID)
+#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX) || \
+    BUILDFLAG(IS_CHROMEOS)
 // Reads favicons for the IWA represented by `page_url` in all available sizes
 // from the disk.
 // `callback` is always run asynchronously for consistency with how extensions
