diff --git chrome/browser/BUILD.gn chrome/browser/BUILD.gn
--- chrome/browser/BUILD.gn
+++ chrome/browser/BUILD.gn
@@ -6911,6 +6911,13 @@ static_library("browser") {
     ]
   }
 
+  if (is_qnx) {
+    sources += [
+      "policy/browser_dm_token_storage_linux.cc",
+      "policy/browser_dm_token_storage_linux.h",
+    ]
+  }
+
   if (is_linux || is_chromeos) {
     sources += [
       "chrome_browser_main_linux.cc",
diff --git chrome/browser/policy/chrome_browser_cloud_management_controller_desktop.cc chrome/browser/policy/chrome_browser_cloud_management_controller_desktop.cc
--- chrome/browser/policy/chrome_browser_cloud_management_controller_desktop.cc
+++ chrome/browser/policy/chrome_browser_cloud_management_controller_desktop.cc
@@ -59,9 +59,9 @@
 #include "chrome/browser/policy/browser_dm_token_storage_mac.h"
 #endif  // BUILDFLAG(IS_MAC)
 
-#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
+#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_QNX)
 #include "chrome/browser/policy/browser_dm_token_storage_linux.h"
-#endif  // BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
+#endif  // BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_QNX)
 
 #if BUILDFLAG(IS_WIN)
 #include "chrome/browser/policy/browser_dm_token_storage_win.h"
@@ -115,7 +115,7 @@ void ChromeBrowserCloudManagementControllerDesktop::
 
 #if BUILDFLAG(IS_MAC)
   storage_delegate = std::make_unique<BrowserDMTokenStorageMac>();
-#elif BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
+#elif BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_QNX)
   storage_delegate = std::make_unique<BrowserDMTokenStorageLinux>();
 #elif BUILDFLAG(IS_WIN)
   storage_delegate = std::make_unique<BrowserDMTokenStorageWin>();
@@ -269,12 +269,12 @@ ChromeBrowserCloudManagementControllerDesktop::GetReportingDelegateFactory() {
 std::unique_ptr<enterprise_reporting::SaasUsageReportingDelegateFactory>
 ChromeBrowserCloudManagementControllerDesktop::
     GetSaasUsageReportingDelegateFactory() {
-#if BUILDFLAG(IS_CHROMEOS)
-  return nullptr;
-#else
+#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_MAC)
   return enterprise_reporting::SaasUsageReportingDelegateFactoryDesktop::
       CreateForBrowser();
-#endif  // !BUILDFLAG(IS_CHROMEOS)
+#else
+  return nullptr;
+#endif  // BUILDFLAG(IS_WIN) || BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_MAC)
 }
 
 void ChromeBrowserCloudManagementControllerDesktop::SetGaiaURLLoaderFactory(
