diff --git chrome/browser/ui/BUILD.gn chrome/browser/ui/BUILD.gn
--- chrome/browser/ui/BUILD.gn
+++ chrome/browser/ui/BUILD.gn
@@ -5075,7 +5075,10 @@
     }
 
     if (is_qnx) {
-      sources += [ "views/chrome_views_delegate_qnx.cc" ]
+      sources += [
+        "views/chrome_views_delegate_qnx.cc",
+        "views/frame/browser_desktop_window_tree_host_qnx.cc",
+        "views/frame/browser_native_widget_factory_qnx.cc",
+      ]
     }
     if (is_qnx) {
       sources += [ "views/status_icons/status_tray_qnx.cc" ]
