diff --git chrome/browser/ui/tab_contents/BUILD.gn chrome/browser/ui/tab_contents/BUILD.gn
--- chrome/browser/ui/tab_contents/BUILD.gn
+++ chrome/browser/ui/tab_contents/BUILD.gn
@@ -51,7 +51,7 @@ source_set("impl") {
     sources += [ "tab_contents_iterator.cc" ]
   }
 
-  if (is_win || is_mac || is_linux || is_chromeos) {
+  if (is_win || is_mac || is_linux || is_chromeos || is_qnx) {
     sources += [
       "chrome_web_contents_menu_helper.cc",
       "chrome_web_contents_view_handle_drop.cc",
