diff --git chrome/browser/BUILD.gn chrome/browser/BUILD.gn
index 969c2af4439e4..df1db5a92f6d5 100644
--- chrome/browser/BUILD.gn
+++ chrome/browser/BUILD.gn
@@ -6988,8 +6988,12 @@ static_library("browser") {
       "policy/messaging_layer/util/manual_test_heartbeat_event.h",
       "policy/messaging_layer/util/manual_test_heartbeat_event_factory.cc",
       "policy/messaging_layer/util/manual_test_heartbeat_event_factory.h",
-      "renderer_context_menu/spelling_options_submenu_observer.cc",
-      "renderer_context_menu/spelling_options_submenu_observer.h",
+    ]
+  }
+
+  if (is_qnx) {
+    sources += [
+      "renderer_context_menu/spelling_options_submenu_observer_qnx.cc",
     ]
   }
 
diff --git chrome/browser/renderer_context_menu/spelling_options_submenu_observer.h chrome/browser/renderer_context_menu/spelling_options_submenu_observer.h
index 4954048e718931075079e74a76ec70cd78239dfd..ee3e3f360af377659bfbda473510b3461693f4f3 100644
--- chrome/browser/renderer_context_menu/spelling_options_submenu_observer.h
+++ chrome/browser/renderer_context_menu/spelling_options_submenu_observer.h
@@ -9,6 +9,7 @@
 #include <vector>
 
 #include "base/memory/raw_ptr.h"
+#include "build/build_config.h"
 #include "chrome/browser/spellchecker/spellcheck_service.h"
 #include "components/prefs/pref_member.h"
 #include "components/renderer_context_menu/render_view_context_menu_observer.h"
@@ -46,6 +47,7 @@ class SpellingOptionsSubMenuObserver : public RenderViewContextMenuObserver {
   // submenu and adds it to the parent menu.
   ui::SimpleMenuModel submenu_model_;
 
+#if !BUILDFLAG(IS_QNX)
   // The ID of radio items representing languages available for spellchecking.
   int language_group_id_;
 
@@ -57,6 +59,7 @@ class SpellingOptionsSubMenuObserver : public RenderViewContextMenuObserver {
 
   // Flag indicating whether the server-powered spellcheck service is enabled.
   BooleanPrefMember use_spelling_service_;
+#endif
 };
 
 #endif  // CHROME_BROWSER_RENDERER_CONTEXT_MENU_SPELLING_OPTIONS_SUBMENU_OBSERVER_H_
