diff --git chrome/browser/resources/settings/BUILD.gn chrome/browser/resources/settings/BUILD.gn
index da914af0eb958dce35250b05e2bf01388bd40f5f..0dce5cc1a6d9c567ea2f29c06214a8a5f5fa5e11 100644
--- chrome/browser/resources/settings/BUILD.gn
+++ chrome/browser/resources/settings/BUILD.gn
@@ -294,7 +294,7 @@ build_webui("build") {
-  if (is_linux) {
+  if (is_linux || is_qnx) {
     web_component_files += [ "a11y_page/captions_page.ts" ]
   }
 
@@ -306,7 +306,7 @@ build_webui("build") {
-  if (is_win || is_mac || is_linux) {
+  if (is_win || is_mac || is_linux || is_qnx) {
     web_component_files += [ "a11y_page/ax_annotations_section.ts" ]
   }
