diff --git chrome/browser/sync/BUILD.gn chrome/browser/sync/BUILD.gn
index 84dc458d1e3d8..a310689c4fe53 100644
--- chrome/browser/sync/BUILD.gn
+++ chrome/browser/sync/BUILD.gn
@@ -6,7 +6,7 @@ import("//chrome/common/features.gni")
 import("//components/supervised_user/buildflags.gni")
 import("//extensions/buildflags/buildflags.gni")
 
-assert(is_win || is_mac || is_linux || is_chromeos || is_android)
+assert(is_win || is_mac || is_linux || is_chromeos || is_android || is_qnx)
 
 source_set("sync") {
   sources = [
