diff --git components/storage_monitor/BUILD.gn components/storage_monitor/BUILD.gn
--- components/storage_monitor/BUILD.gn
+++ components/storage_monitor/BUILD.gn
@@ -70,15 +70,19 @@ component("storage_monitor") {
       sources += [
         "mtab_watcher_linux.cc",
         "mtab_watcher_linux.h",
         "storage_monitor_linux.cc",
         "storage_monitor_linux.h",
       ]
     } else {
       sources += [ "storage_monitor_dummy.cc" ]
     }
   }

+  if (is_qnx) {
+    sources += [ "storage_monitor_linux_qnx.cc" ]
+  }
+
   if (use_udev) {
-    if (is_linux || is_chromeos) {
+    if (is_linux || is_chromeos || is_qnx) {
       sources += [
         "udev_util_linux.cc",
         "udev_util_linux.h",
