diff --git content/browser/scheduler/responsiveness/native_event_observer.cc content/browser/scheduler/responsiveness/native_event_observer.cc
index 6ae364319a673492786ea71538c8847bc20a17d8..8797807a2ddadcc9fe517b398e67cff3e4c374c1 100644
--- content/browser/scheduler/responsiveness/native_event_observer.cc
+++ content/browser/scheduler/responsiveness/native_event_observer.cc
@@ -15,10 +15,6 @@
 
 #include "ui/events/platform/platform_event_source.h"
 
-#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
-#include "ui/events/platform/platform_event_source.h"
-#endif
-
 #if BUILDFLAG(IS_WIN)
 #include "base/task/current_thread.h"
 #endif
@@ -38,7 +34,7 @@ NativeEventObserver::~NativeEventObserver() {
   DeregisterObserver();
 }
 
-#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
+#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_QNX)
 void NativeEventObserver::RegisterObserver() {
   CHECK(ui::PlatformEventSource::GetInstance());
   ui::PlatformEventSource::GetInstance()->AddPlatformEventObserver(this);
@@ -67,7 +63,8 @@ void NativeEventObserver::PlatformEventSourceDestroying() {
   DeregisterObserver();
 }
 
-#endif  // BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
+#endif  // BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) || \
+        // BUILDFLAG(IS_QNX)
 
 #if BUILDFLAG(IS_WIN)
 void NativeEventObserver::RegisterObserver() {
diff --git content/browser/scheduler/responsiveness/native_event_observer.h content/browser/scheduler/responsiveness/native_event_observer.h
index 9c0226c82be4e8dd26301c3c45e568df25806e14..30e2ad1e85af6f221739ce91c910b501dcc05685 100644
--- content/browser/scheduler/responsiveness/native_event_observer.h
+++ content/browser/scheduler/responsiveness/native_event_observer.h
@@ -16,7 +16,7 @@
 #include "content/public/browser/native_event_processor_observer_mac.h"
 #endif
 
-#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
+#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_QNX)
 #include "ui/events/platform/platform_event_observer.h"
 #endif
 
@@ -41,7 +41,7 @@ namespace responsiveness {
 class CONTENT_EXPORT NativeEventObserver
 #if BUILDFLAG(IS_MAC)
     : public NativeEventProcessorObserver
-#elif BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
+#elif BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_QNX)
     : public ui::PlatformEventObserver
 #elif BUILDFLAG(IS_WIN)
     : public base::MessagePumpForUI::Observer
@@ -58,7 +58,7 @@ class CONTENT_EXPORT NativeEventObserver
   NativeEventObserver(WillRunEventCallback will_run_event_callback,
                       DidRunEventCallback did_run_event_callback);
 
-#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
+#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_QNX)
 
   NativeEventObserver(const NativeEventObserver&) = delete;
   NativeEventObserver& operator=(const NativeEventObserver&) = delete;
@@ -74,7 +74,7 @@ class CONTENT_EXPORT NativeEventObserver
   // Exposed for tests.
   void WillRunNativeEvent(const void* opaque_identifier) override;
   void DidRunNativeEvent(const void* opaque_identifier) override;
-#elif BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
+#elif BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_QNX)
   // ui::PlatformEventObserver overrides:
   void WillProcessEvent(const ui::PlatformEvent& event) override;
   void DidProcessEvent(const ui::PlatformEvent& event) override;
@@ -89,7 +89,7 @@ class CONTENT_EXPORT NativeEventObserver
   void RegisterObserver();
   void DeregisterObserver();
 
-#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
+#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_QNX)
   struct EventInfo {
     raw_ptr<const void> unique_id;
   };
   std::vector<EventInfo> events_being_processed_;
#endif
diff --git content/common/BUILD.gn content/common/BUILD.gn
index 463f7160eb5e100f93d9c98b2bb6cf2a07e72a76..5b8b4e7db812fb0ecad6fb6e488a19f6e1b9f4f1 100644
--- content/common/BUILD.gn
+++ content/common/BUILD.gn
@@ -396,6 +396,10 @@ source_set("common") {
     sources -= [ "font_list_fontconfig.cc" ]
   }
 
+  if (is_qnx) {
+    sources += [ "font_list_qnx.cc" ]
+  }
+
   if (is_mac) {
     deps += [ "//media/gpu" ]
   }
diff --git content/renderer/BUILD.gn content/renderer/BUILD.gn
index bbb6f87c5b824b6899b07de1a0e202903c2c983a..8f8ebad8ca7b7db95fc96d099f1bf8d7f1321f26 100644
--- content/renderer/BUILD.gn
+++ content/renderer/BUILD.gn
@@ -196,6 +196,10 @@ target(link_target_type, "renderer") {
     sources += [ "renderer_main_platform_delegate_linux.cc" ]
   }
 
+  if (is_qnx) {
+    sources += [ "renderer_main_platform_delegate_qnx.cc" ]
+  }
+
   if (is_mac) {
     sources += [
       "renderer_main_platform_delegate_mac.mm",
