diff --git third_party/crashpad/crashpad/handler/BUILD.gn third_party/crashpad/crashpad/handler/BUILD.gn
--- third_party/crashpad/crashpad/handler/BUILD.gn
+++ third_party/crashpad/crashpad/handler/BUILD.gn
@@ -93,6 +93,10 @@ if (!crashpad_is_ios) {
       ]
     }
 
+    if (is_qnx) {
+      sources += [ "qnx/noop_client_stubs.cc" ]
+    }
+
     public_configs = [ "..:crashpad_config" ]
 
     public_deps = [
