--- ui/message_center/dummy_message_center.cc
+++ ui/message_center/dummy_message_center.cc
@@ -9,8 +9,8 @@
 // and link with Android implementations of Chrome which do not have
 // notification systems yet. This is to avoid spreading compile-time flags
 // everywhere in the code.
-#if !(BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_FUCHSIA))
-#error This file should only be used in Android and Fuchsia.
+#if !(BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_FUCHSIA) || BUILDFLAG(IS_QNX))
+#error This file should only be used in Android, Fuchsia, and QNX.
 #endif
 
 namespace message_center {
