diff --git device/fido/hid/fido_hid_packet.cc device/fido/hid/fido_hid_packet.cc
index 1c2351d1c464e94321546103b7641455d7ce9e17..e766bd83d0c656383b6f39e8ca18d6c826904483 100644
--- device/fido/hid/fido_hid_packet.cc
+++ device/fido/hid/fido_hid_packet.cc
@@ -34,7 +34,8 @@ std::unique_ptr<FidoHidInitPacket> FidoHidInitPacket::CreateFromSerializedData(
   channel_id |= serialized[index++] & 0xff;
 
   auto command = static_cast<FidoHidDeviceCommand>(serialized[index++] & 0x7f);
-  if (!std::ranges::contains(GetFidoHidDeviceCommandList(), command)) {
+  const auto device_commands = GetFidoHidDeviceCommandList();
+  if (!std::ranges::contains(device_commands, command)) {
     return nullptr;
   }
 
