diff --git gpu/ipc/service/gpu_init.cc gpu/ipc/service/gpu_init.cc
index 08e9fbc34133fa4b44dfda243de8e2c8e944758d..3f07b494fe257995f7f9c0f822ff250d5a580ddd 100644
--- gpu/ipc/service/gpu_init.cc
+++ gpu/ipc/service/gpu_init.cc
@@ -960,9 +960,12 @@ bool GpuInit::InitializeAndStartSandbox(base::CommandLine* command_line,
   SetSkiaBackendType();
 #if BUILDFLAG(IS_OZONE)
   ui::OzonePlatform::GetInstance()->AfterSandboxEntry();
-  [[maybe_unused]] auto* factory =
+#if BUILDFLAG(ENABLE_VULKAN) || \
+    (BUILDFLAG(SKIA_USE_DAWN) && BUILDFLAG(IS_CHROMEOS))
+  auto* factory =
       ui::OzonePlatform::GetInstance()->GetSurfaceFactoryOzone();
   bool filter_set = false;
+#endif
 #if BUILDFLAG(ENABLE_VULKAN)
   if (gpu_feature_info_.status_values[GPU_FEATURE_TYPE_VULKAN] ==
           kGpuFeatureStatusEnabled &&
