
diff --git components/gwp_asan/crash_handler/crash_analyzer.cc components/gwp_asan/crash_handler/crash_analyzer.cc
index 438271a53d48e..e090fc14d7536 100644
--- components/gwp_asan/crash_handler/crash_analyzer.cc
+++ components/gwp_asan/crash_handler/crash_analyzer.cc
@@ -114,7 +114,8 @@ bool CrashAnalyzer::GetExceptionInfo(
 
 crashpad::VMAddress CrashAnalyzer::GetAccessAddress(
     const crashpad::ExceptionSnapshot& exception) {
-#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_ANDROID)
+#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_ANDROID) || \
+    BUILDFLAG(IS_QNX)
   if (exception.Exception() == SIGSEGV || exception.Exception() == SIGBUS)
     return exception.ExceptionAddress();
 #elif BUILDFLAG(IS_APPLE)
@@ -323,6 +324,8 @@ bool CrashAnalyzer::AnalyzeLightweightDetectorCrash(
       // https://elixir.bootlin.com/linux/v6.2.2/source/arch/x86/kernel/traps.c#L719
       exception->Exception() == SIGSEGV &&
       exception->ExceptionInfo() == SI_KERNEL
+#elif BUILDFLAG(IS_QNX)
+      false
 #elif BUILDFLAG(IS_APPLE)
       // https://opensource.apple.com/source/xnu/xnu-1699.24.8/osfmk/i386/trap.c
       exception->Exception() == EXC_BAD_ACCESS &&

