diff --git chrome/browser/BUILD.gn chrome/browser/BUILD.gn
--- chrome/browser/BUILD.gn
+++ chrome/browser/BUILD.gn
@@ -6547,10 +6547,12 @@
   }
 
   if (enterprise_watermark) {
-    sources += [
-      "enterprise/watermark/settings.cc",
-      "enterprise/watermark/settings.h",
-    ]
+    if (!is_qnx) {
+      sources += [
+        "enterprise/watermark/settings.cc",
+        "enterprise/watermark/settings.h",
+      ]
+    }
     deps += [
       "//chrome/browser/enterprise/watermark:watermark_view_lib",
       "//chrome/browser/ui/webui/watermark",
diff --git chrome/browser/ui/BUILD.gn chrome/browser/ui/BUILD.gn
--- chrome/browser/ui/BUILD.gn
+++ chrome/browser/ui/BUILD.gn
@@ -1434,6 +1434,13 @@
       "webui/webui_util_desktop.h",
     ]
 
+    if (is_qnx) {
+      sources -= [
+        "read_anything/read_anything_prefs.cc",
+        "read_anything/read_anything_prefs.h",
+      ]
+    }
+
     if (is_chrome_branded) {
       sources += [
         "webui/media_router/cast_feedback_ui.cc",
@@ -2949,8 +2956,6 @@
 
   if (is_win || is_mac || is_linux || is_chromeos || is_qnx) {
     sources += [
-      "frame/window_frame_util.cc",
-      "frame/window_frame_util.h",
       "incognito_clear_browsing_data_dialog_interface.h",
       "views/autofill/payments/webauthn_dialog_view.cc",
       "views/autofill/payments/webauthn_dialog_view.h",
@@ -2969,6 +2974,14 @@
       "views/profiles/signin_view_controller_delegate_views.cc",
       "views/profiles/signin_view_controller_delegate_views.h",
     ]
+
+    if (!is_qnx) {
+      sources += [
+        "frame/window_frame_util.cc",
+        "frame/window_frame_util.h",
+      ]
+    }
+
     deps += [
       "//chrome/browser/ui/webui/signin/history_sync_optin:mojo_bindings",
       "//chrome/browser/contextual_tasks:ui",
diff --git components/enterprise/BUILD.gn components/enterprise/BUILD.gn
--- components/enterprise/BUILD.gn
+++ components/enterprise/BUILD.gn
@@ -77,7 +77,7 @@
     ]
   }
 
-  if (!is_ios && !is_android) {
+  if (!is_ios && !is_android && !is_qnx) {
     sources += [
       "browser/promotion/promotion_eligibility_checker.cc",
       "browser/promotion/promotion_eligibility_checker.h",
