diff --git chrome/browser/accessibility/live_translate_controller_factory.cc chrome/browser/accessibility/live_translate_controller_factory.cc
index 0ec2fb4423ff31db3a34de8bd084fc7ec7c37822..ce222e8c5b55ea0e04d7d36b43ba133f440ec799 100644
--- chrome/browser/accessibility/live_translate_controller_factory.cc
+++ chrome/browser/accessibility/live_translate_controller_factory.cc
@@ -6,16 +6,19 @@
 
 #include "base/feature_list.h"
 #include "base/no_destructor.h"
-#include "chrome/browser/on_device_translation/service_controller_manager_factory.h"
 #include "chrome/browser/profiles/profile.h"
 #include "components/live_caption/features.h"
 #include "components/live_caption/google_api_translation_dispatcher.h"
 #include "components/live_caption/live_translate_controller.h"
-#include "components/live_caption/translation_dispatcher_on_device.h"
 #include "components/on_device_translation/buildflags/buildflags.h"
+#include "google_apis/google_api_keys.h"
+
+#if BUILDFLAG(ENABLE_ON_DEVICE_TRANSLATION)
+#include "chrome/browser/on_device_translation/service_controller_manager_factory.h"
+#include "components/live_caption/translation_dispatcher_on_device.h"
 #include "components/on_device_translation/service_controller.h"
 #include "components/on_device_translation/service_controller_manager.h"
-#include "google_apis/google_api_keys.h"
+#endif
 
 namespace captions {
 
@@ -60,6 +63,7 @@ LiveTranslateControllerFactory::BuildServiceInstanceForBrowserContext(
   std::unique_ptr<TranslationDispatcher> on_device_dispatcher;
   std::unique_ptr<TranslationDispatcher> google_api_dispatcher;
 
+#if BUILDFLAG(ENABLE_ON_DEVICE_TRANSLATION)
   // Only set on_device_dispatcher if feature flag is set.
   if (base::FeatureList::IsEnabled(
           live_caption::kLiveCaptionOnDeviceTranslation)) {
@@ -68,6 +72,7 @@ LiveTranslateControllerFactory::BuildServiceInstanceForBrowserContext(
             on_device_translation::OnDeviceTranslationServiceController>(
             profile->GetPrefs(), ""));
   }
+#endif
   google_api_dispatcher = std::make_unique<GoogleApiTranslationDispatcher>(
       google_apis::GetAPIKey(), context);
 
