diff --git media/audio/BUILD.gn media/audio/BUILD.gn
--- media/audio/BUILD.gn
+++ media/audio/BUILD.gn
@@ -291,7 +291,7 @@ source_set("audio") {
     deps += [ "//media/base/android:media_jni_headers" ]
   }
 
-  if (is_linux || is_chromeos) {
+  if ((is_linux || is_chromeos) && !is_qnx) {
     sources += [ "linux/audio_manager_linux.cc" ]
   }
 
diff --git media/capture/BUILD.gn media/capture/BUILD.gn
--- media/capture/BUILD.gn
+++ media/capture/BUILD.gn
@@ -397,7 +397,7 @@ source_set("capture_gpu_channel") {
   deps = []
 
   # Establish GPU Channel
-  if (enable_gpu_channel_media_capture) {
+  if (enable_gpu_channel_media_capture && !is_qnx) {
     sources += [
       "video/video_capture_gpu_channel_host.cc",
       "video/video_capture_gpu_channel_host.h",
@@ -409,7 +409,10 @@ source_set("capture_gpu_channel") {
   }
 
   if (is_qnx) {
-    sources += [ "video/video_capture_gpu_channel_host_qnx.cc" ]
+    sources += [
+      "video/video_capture_gpu_channel_host.h",
+      "video/video_capture_gpu_channel_host_qnx.cc",
+    ]
     deps += [
       "//base",
       "//gpu/command_buffer/client",
diff --git media/capture/video/video_capture_gpu_channel_host_qnx.cc media/capture/video/video_capture_gpu_channel_host_qnx.cc
--- media/capture/video/video_capture_gpu_channel_host_qnx.cc
+++ media/capture/video/video_capture_gpu_channel_host_qnx.cc
@@ -22,6 +22,9 @@ VideoCaptureGpuChannelHost::GetSharedImageInterface() {
   return nullptr;
 }
 
+void VideoCaptureGpuChannelHost::SetSharedImageInterface(
+    scoped_refptr<gpu::SharedImageInterface> shared_image_interface) {}
+
 void VideoCaptureGpuChannelHost::OnContextLost() {}
 
 void VideoCaptureGpuChannelHost::AddObserver(
