diff --git media/capture/BUILD.gn media/capture/BUILD.gn
--- media/capture/BUILD.gn
+++ media/capture/BUILD.gn
@@ -237,7 +237,7 @@ component("capture_lib") {
     public_deps += [ "video/win" ]
   }
 
-  if (is_linux || is_chromeos) {
+  if ((is_linux || is_chromeos) && !is_qnx) {
     sources += [
       "video/linux/scoped_v4l2_device_fd.cc",
       "video/linux/scoped_v4l2_device_fd.h",
@@ -264,7 +264,7 @@ component("capture_lib") {
     deps += [ "//gpu/command_buffer/client" ]
   }
 
-  if (is_linux) {
+  if (is_linux && !is_qnx) {
     sources += [
       "video/linux/v4l2_capture_delegate_gpu_helper.cc",
       "video/linux/v4l2_capture_delegate_gpu_helper.h",
@@ -445,7 +445,7 @@ source_set("test_support") {
     "//ui/gfx:test_support",
   ]
 
-  if (is_linux) {
+  if (is_linux && !is_qnx) {
     sources += [
       "video/linux/fake_device_provider.cc",
       "video/linux/fake_device_provider.h",
@@ -517,14 +517,14 @@ test("capture_unittests") {
     "//ui/gfx:test_support",
   ]
 
-  if (is_linux) {
+  if (is_linux && !is_qnx) {
     sources += [
       "video/linux/v4l2_capture_delegate_unittest.cc",
       "video/linux/video_capture_device_factory_v4l2_unittest.cc",
     ]
   }
 
-  if (is_linux) {
+  if (is_linux && !is_qnx) {
     sources += [
       "video/linux/v4l2_capture_delegate_gpu_helper_unittest.cc",
       "video/linux/v4l2_gpu_memory_buffer_tracker_unittest.cc",
