diff --git net/base/network_interfaces_posix.cc net/base/network_interfaces_posix.cc
index 0f330e98d31d3c44877a7f5fa96d4999af273430..09909fe87740240eafe48e24e72cbafff75b4fd5 100644
--- net/base/network_interfaces_posix.cc
+++ net/base/network_interfaces_posix.cc
@@ -5,6 +5,7 @@
 #include "net/base/network_interfaces_posix.h"
 
 #include <netinet/in.h>
+#include <sys/socket.h>
 #include <sys/types.h>
 
 #include <memory>
diff --git net/socket/udp_socket_posix.cc net/socket/udp_socket_posix.cc
index f3cdb1e050c0b25f4109891ab0d472afb2753f07..d0c200ef43ca7aac751b4be21789d140ce09e66f 100644
--- net/socket/udp_socket_posix.cc
+++ net/socket/udp_socket_posix.cc
@@ -679,13 +679,15 @@ int UDPSocketPosix::SetRecvTos() {
 }
 
 void UDPSocketPosix::SetMsgConfirm(bool confirm) {
-#if !BUILDFLAG(IS_APPLE)
+#if !BUILDFLAG(IS_APPLE) && defined(MSG_CONFIRM)
   if (confirm) {
     sendto_flags_ |= MSG_CONFIRM;
   } else {
     sendto_flags_ &= ~MSG_CONFIRM;
   }
-#endif  // !BUILDFLAG(IS_APPLE)
+#else
+  (void)confirm;
+#endif  // !BUILDFLAG(IS_APPLE) && defined(MSG_CONFIRM)
 }
 
 int UDPSocketPosix::AllowAddressReuse() {
