diff --git ui/ozone/BUILD.gn ui/ozone/BUILD.gn
--- ui/ozone/BUILD.gn
+++ ui/ozone/BUILD.gn
@@ -71,6 +71,11 @@ if (ozone_platform_flatland) {
   ozone_platform_test_deps += [ "platform/flatland:flatland_unittests" ]
 }
 
+if (ozone_platform_qnx && is_qnx) {
+  ozone_platforms += [ "qnx" ]
+  ozone_platform_deps += [ "platform/qnx" ]
+}
+
 platform_list_cc_file = "$target_gen_dir/platform_list.cc"
 platform_list_h_file = "$target_gen_dir/platform_list.h"
 platform_list_txt_file = "$target_gen_dir/platform_list.txt"
