diff --git chrome/browser/webauthn/password_credential_ui_controller.cc chrome/browser/webauthn/password_credential_ui_controller.cc
index 3f903bb0292d2acca11ee9c8f5216872a86c81a1..f5643b7fdc1797c786422c3b0e129f417e7e82d7 100644
--- chrome/browser/webauthn/password_credential_ui_controller.cc
+++ chrome/browser/webauthn/password_credential_ui_controller.cc
@@ -26,12 +26,12 @@ using content::WebContents;
 namespace {
 
 std::u16string GetAuthenticationMessage(std::string_view rp_id) {
-#if BUILDFLAG(IS_LINUX)
+#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_QNX)
   return u"";
 #else
   return l10n_util::GetStringFUTF16(IDS_PASSWORD_MANAGER_FILLING_REAUTH,
                                     base::UTF8ToUTF16(rp_id));
-#endif  // BUILDFLAG(IS_LINUX)
+#endif  // BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_QNX)
 }
 
 }  // namespace
