diff --git components/password_manager/core/common/password_manager_pref_names.h components/password_manager/core/common/password_manager_pref_names.h
--- components/password_manager/core/common/password_manager_pref_names.h
+++ components/password_manager/core/common/password_manager_pref_names.h
@@ -287,13 +287,13 @@ inline constexpr char kLastNegativePasswordChangeTimestamp[] =
     "password_manager.last_negative_password_change_timestamp";
 
 #if BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_WIN) || \
-    BUILDFLAG(IS_MAC)
+    BUILDFLAG(IS_MAC) || BUILDFLAG(IS_QNX)
 // List of urls for which password manager is disabled/blocked.
 inline constexpr char kPasswordManagerBlocklist[] =
     "password_manager.password_manager_blocklist";
 #endif  // BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_WIN) ||
-        // BUILDFLAG(IS_MAC)
+        // BUILDFLAG(IS_MAC) || BUILDFLAG(IS_QNX)
 
 #if BUILDFLAG(IS_CHROMEOS)
 inline constexpr char kPinAuthenticationAvailableOnChromeOS[] =
