diff --git components/password_manager/core/browser/password_manager_client.h components/password_manager/core/browser/password_manager_client.h
index cb9b7cf510515ab7be7c7cd5f2b75bc6975aa432..340533fee76e97de4cbb43b749820908627f1609 100644
--- components/password_manager/core/browser/password_manager_client.h
+++ components/password_manager/core/browser/password_manager_client.h
@@ -544,7 +544,7 @@ class PasswordManagerClient {
   virtual void TriggerSignIn(signin_metrics::AccessPoint access_point) const;
 
 #if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_MAC) || \
-    BUILDFLAG(IS_CHROMEOS)
+    BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_QNX)
   // Shows the bubble with the details of the `form`.
   virtual void OpenPasswordDetailsBubble(
       const password_manager::PasswordForm& form) = 0;
@@ -554,7 +554,7 @@ class PasswordManagerClient {
   virtual void MaybeShowSavePasswordPrimingPromo(const GURL& current_url) = 0;
 
 #endif  // BUILDFLAG(IS_WIN) || BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_MAC) ||
-        // BUILDFLAG(IS_CHROMEOS)
+        // BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_QNX)
 
 #if !BUILDFLAG(IS_IOS)
   // Creates and show the cross domain confirmation popup.
diff --git components/password_manager/core/browser/stub_password_manager_client.cc components/password_manager/core/browser/stub_password_manager_client.cc
index 8f9f3edf5fb99ff7427294138e935bb541a87f93..8b55dfc337183da4a9c30d00d41056472c48581a 100644
--- components/password_manager/core/browser/stub_password_manager_client.cc
+++ components/password_manager/core/browser/stub_password_manager_client.cc
@@ -226,13 +226,13 @@ version_info::Channel StubPasswordManagerClient::GetChannel() const {
 }
 
 #if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_MAC) || \
-    BUILDFLAG(IS_CHROMEOS)
+    BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_QNX)
 void StubPasswordManagerClient::OpenPasswordDetailsBubble(
     const password_manager::PasswordForm& form) {}
 void StubPasswordManagerClient::MaybeShowSavePasswordPrimingPromo(
     const GURL& current_url) {}
 #endif  // BUILDFLAG(IS_WIN) || BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_MAC) ||
-        // BUILDFLAG(IS_CHROMEOS)
+        // BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_QNX)
 
 #if !BUILDFLAG(IS_IOS)
 std::unique_ptr<
diff --git components/password_manager/core/browser/stub_password_manager_client.h components/password_manager/core/browser/stub_password_manager_client.h
index 8dd0a81f071f585e07b87f6e81f03e4e922f3d6b..9a3ea8405517d6239dc795dbf0a56324d4534743 100644
--- components/password_manager/core/browser/stub_password_manager_client.h
+++ components/password_manager/core/browser/stub_password_manager_client.h
@@ -84,12 +84,12 @@ class StubPasswordManagerClient : public PasswordManagerClient {
   MockPasswordFeatureManager* GetPasswordFeatureManager();
   version_info::Channel GetChannel() const override;
 #if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_MAC) || \
-    BUILDFLAG(IS_CHROMEOS)
+    BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_QNX)
   void OpenPasswordDetailsBubble(
       const password_manager::PasswordForm& form) override;
   void MaybeShowSavePasswordPrimingPromo(const GURL& current_url) override;
 #endif  // BUILDFLAG(IS_WIN) || BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_MAC) ||
-        // BUILDFLAG(IS_CHROMEOS)
+        // BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_QNX)
 #if !BUILDFLAG(IS_IOS)
   std::unique_ptr<
       password_manager::PasswordCrossDomainConfirmationPopupController>
