diff --git a/components/enterprise/client_certificates/core/private_key_factory.cc b/components/enterprise/client_certificates/core/private_key_factory.cc
index f2365aaf5ab77..a0f1cd174ebb7 100644
--- a/components/enterprise/client_certificates/core/private_key_factory.cc
+++ b/components/enterprise/client_certificates/core/private_key_factory.cc
@@ -122,11 +122,13 @@ void PrivateKeyFactoryImpl::OnPrivateKeyCreated(
     PrivateKeyCallback callback,
     scoped_refptr<PrivateKey> private_key) {
   if (!private_key && source != PrivateKeySource::kSoftwareKey) {
-    for (auto fallback_source =
-             ++std::find(std::begin(kKeySourcesOrderedBySecurity),
-                         std::end(kKeySourcesOrderedBySecurity), source);
-         fallback_source != std::end(kKeySourcesOrderedBySecurity);
-         fallback_source++) {
+    for (auto it = std::find(std::begin(kKeySourcesOrderedBySecurity),
+                           std::end(kKeySourcesOrderedBySecurity), source);
+         it != std::end(kKeySourcesOrderedBySecurity);
+         ++it) {
+      auto fallback_source = std::next(it);
+      if (fallback_source == std::end(kKeySourcesOrderedBySecurity))
+        break;
       auto it = sub_factories_.find(*fallback_source);
       if (it != sub_factories_.end()) {
         // If a more secure key failed to be created, fallback to creating a
