diff --git chrome/browser/profiles/chrome_browser_main_extra_parts_profiles.cc chrome/browser/profiles/chrome_browser_main_extra_parts_profiles.cc
index b183333a164c2906366591e5eac65963421a00fa..8ecd53578b404318bfa5c5a8e7b866870867eeaf 100644
--- chrome/browser/profiles/chrome_browser_main_extra_parts_profiles.cc
+++ chrome/browser/profiles/chrome_browser_main_extra_parts_profiles.cc
@@ -396,7 +396,7 @@
 #endif
 
 #if BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_MAC) || \
-    BUILDFLAG(IS_WIN)
+    BUILDFLAG(IS_WIN) || BUILDFLAG(IS_QNX)
 #include "chrome/browser/enterprise/client_certificates/certificate_provisioning_service_factory.h"
 #include "chrome/browser/enterprise/client_certificates/certificate_store_factory.h"
 #include "chrome/browser/enterprise/idle/idle_service_factory.h"
