diff --git chrome/browser/net/profile_network_context_service.cc chrome/browser/net/profile_network_context_service.cc
index 226a3863950c14af54207906899008d2491f8a27..de564039df511e52bb819fd598e97e27f2945f9a 100644
--- chrome/browser/net/profile_network_context_service.cc
+++ chrome/browser/net/profile_network_context_service.cc
@@ -136,9 +136,9 @@
 #include "net/ssl/client_cert_store_mac.h"
 #endif  // BUILDFLAG(IS_MAC)
 
-#if BUILDFLAG(IS_ANDROID)
+#if BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_QNX)
 #include "net/ssl/client_cert_store_empty.h"
-#endif  // BUILDFLAG(IS_ANDROID)
+#endif  // BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_QNX)
 
 #if BUILDFLAG(ENABLE_EXTENSIONS)
 #include "extensions/common/constants.h"
@@ -1287,6 +1287,8 @@ ProfileNetworkContextService::CreateClientCertStore() {
   // the OS as part of the call to show the cert selection dialog.
   return GetWrappedCertStore(profile_,
                              std::make_unique<net::ClientCertStoreEmpty>());
+#elif BUILDFLAG(IS_QNX)
+  return std::make_unique<net::ClientCertStoreEmpty>();
 #else
 #error Unknown platform.
 #endif
@@ -1306,7 +1308,8 @@ std::vector<uint8_t>
 ProfileNetworkContextService::GetEncryptedCachePrimaryKey() {
   std::string encoded_encrypted_primary_key = profile_->GetPrefs()->GetString(
       enterprise_connectors::kEncryptedCachePrimaryKey);
-  return base::Base64Decode(encoded_encrypted_primary_key).value_or({});
+  return base::Base64Decode(encoded_encrypted_primary_key)
+      .value_or(std::vector<uint8_t>{});
 }
 
 #endif  // BUILDFLAG(ENTERPRISE_CACHE_ENCRYPTION)
