--- cef/patch/qnx/chromium/new_files/ui/ozone/platform/qnx/qnx_gpu_service.cc	2026-07-08 17:01:48.000000000 +0900
+++ ui/ozone/platform/qnx/qnx_gpu_service.cc	2026-07-08 17:01:48.000000000 +0900
@@ -39,7 +39,6 @@
   return base::CommandLine::ForCurrentProcess()->HasSwitch(
       kOzoneQnxGpuTraceSwitch);
 }
-
 }  // namespace
 
 QnxGpuService::QnxGpuService(QnxSurfaceFactoryOzone* surface_factory)

