diff --git a/.gitmodules b/.gitmodules
index d136e5d2532a9..f82970aa8b5e9 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -914,3 +914,6 @@
 	path = webkit/data/test_shell/plugins
 	url = https://chrome-internal.googlesource.com/chrome/data/webkit_plugin_tests
 	gclient-condition = checkout_src_internal
+[submodule "third_party/farmhash_qnx/src"]
+	path = third_party/farmhash_qnx/src
+	url = https://github.com/qnx-ports/farmhash.git
diff --git a/DEPS b/DEPS
index 6fe4536b7acda..6786cba47668e 100644
--- a/DEPS
+++ b/DEPS
@@ -4875,3 +4875,14 @@ recursedeps = [
   # this.
   'src/third_party/catapult',
 ]
+ools-frontend/src',
+  # clank has its own DEPS file, does not need to be in trybot_analyze_config
+  # since the roller does not run tests.
+  'src/clank',
+  'src/components/optimization_guide/internal',
+  'src/ios_internal',
+  # TODO(crbug.com/430571901): Fix telemetry_unittests to use
+  # third_party/webpagereplay instead of the copy inside catapult and remove
+  # this.
+  'src/third_party/catapult',
+]

