diff --git a/.gitmodules b/.gitmodules
index d136e5d2532a9..4a369bbb13984 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -914,3 +914,9 @@
 	path = webkit/data/test_shell/plugins
 	url = https://chrome-internal.googlesource.com/chrome/data/webkit_plugin_tests
 	gclient-condition = checkout_src_internal
+[submodule "third_party/epoll/src"]
+	path = third_party/epoll/src
+	url = https://github.com/qnx-ports/epoll.git
+[submodule "third_party/cpuinfo_qnx/src"]
+	path = third_party/cpuinfo_qnx/src
+	url = https://github.com/qnx-ports/cpuinfo.git
diff --git a/DEPS b/DEPS
index 6fe4536b7acda..8b0b487de4d13 100644
--- a/DEPS
+++ b/DEPS
@@ -2156,6 +2156,12 @@ deps = {
   'src/third_party/emoji-segmenter/src':
     Var('chromium_git') + '/external/github.com/google/emoji-segmenter.git' + '@' + Var('emoji_segmenter_revision'),
 
+  'src/third_party/epoll/src':
+    'https://github.com/qnx-ports/epoll.git' + '@' + 'aea458535243e44627b85b915ccbb658a396e4fb',
+
+  'src/third_party/cpuinfo_qnx/src':
+    'https://github.com/qnx-ports/cpuinfo.git' + '@' + '0cf43cf0ad54f4f8551d5a0c624fe5c4aee64dcc',
+
   'src/third_party/oak/src':
     Var('chromium_git') + '/external/github.com/project-oak/oak.git' + '@' + '96c00a6c99ac382f3f3a8f376bc7a70890d1adaa',

