diff --git chrome/browser/ui/browser_ui_prefs.cc chrome/browser/ui/browser_ui_prefs.cc
index 771d84612a86fba2482b4ecff6f4920522343725..69730dcdc4fe1a2f05720895ea08447766c0a263 100644
--- chrome/browser/ui/browser_ui_prefs.cc
+++ chrome/browser/ui/browser_ui_prefs.cc
@@ -109,9 +109,11 @@ void RegisterBrowserUserPrefs(user_prefs::PrefRegistrySyncable* registry) {
       user_prefs::PrefRegistrySyncable::SYNCABLE_PREF;
 #endif
 
-#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN) || \
+    BUILDFLAG(IS_QNX)
   registry->RegisterIntegerPref(prefs::kSessionRestoreInfoBarTimesShown, 0);
-#endif  // BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN)
+#endif  // BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN) || \
+        // BUILDFLAG(IS_QNX)
 
   registry->RegisterBooleanPref(prefs::kHomePageIsNewTabPage, true,
                                 pref_registration_flags);
diff --git chrome/common/pref_names.h chrome/common/pref_names.h
index 9bb8cc1fe0e7f54d25c5b2deadc41226f36c3753..38665d4c7b6a6641a5ebb3778d5979c86f283316 100644
--- chrome/common/pref_names.h
+++ chrome/common/pref_names.h
@@ -2012,13 +2012,15 @@ inline constexpr char kPinInfoBarTimesShown[] =
     "browser.pin_infobar_times_shown";
 #endif  // BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC)
 
-#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX)
+#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX) || \
+    BUILDFLAG(IS_QNX)
 
 // How many times the session restore infobar has been shown.
 inline constexpr char kSessionRestoreInfoBarTimesShown[] =
     "browser.session_restore_infobar_times_shown";
 
-#endif  // BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX)
+#endif  // BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX) || \
+        // BUILDFLAG(IS_QNX)
 
 // A collection of position, size, and other data relating to the browser
 // window to restore on startup.
