diff --git components/signin/public/base/signin_switches.cc components/signin/public/base/signin_switches.cc
index 149896508f51bd7e3acc7a1c07604c1a7561d350..fcb3c694c5dc87442e3f6f263812968b44e43a92 100644
--- components/signin/public/base/signin_switches.cc
+++ components/signin/public/base/signin_switches.cc
@@ -478,7 +478,8 @@ const base::FeatureParam<int>
         "max_profiles_count_to_show_open_all_button_in_profile_picker", 5};
 #endif  // BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX)
 
-#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX)
+#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX) || \
+    BUILDFLAG(IS_QNX)
 BASE_FEATURE(kPasswordUploadUiUpdate, base::FEATURE_DISABLED_BY_DEFAULT);
 
 BASE_FEATURE(kProfileCreationDeclineSigninCTAExperiment,
@@ -510,7 +511,8 @@ constexpr base::FeatureParam<ProfilePickerVariation>
         &kProfilePickerTextVariations, "profile-picker-variation",
         ProfilePickerVariation::kKeepWorkAndLifeSeparate,
         &kProfilePickerVariations};
-#endif  // BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX)
+#endif  // BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX) || \
+        // BUILDFLAG(IS_QNX)
 
 BASE_FEATURE(kProfilesReordering, base::FEATURE_DISABLED_BY_DEFAULT);
 
diff --git components/signin/public/base/signin_switches.h components/signin/public/base/signin_switches.h
index f974f2ce6597ebb2eb2a49fecfec4417597eaf4a..0558c5bc3e8dd2b805c0e6927ec3932ac678367d 100644
--- components/signin/public/base/signin_switches.h
+++ components/signin/public/base/signin_switches.h
@@ -426,7 +426,8 @@ extern const base::FeatureParam<int>
     kMaxProfilesCountToShowOpenAllButtonInProfilePicker;
 #endif  // BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX)
 
-#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX)
+#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX) || \
+    BUILDFLAG(IS_QNX)
 // Add new entry points for uploading passwords to account storage and update
 // existing ones.
 COMPONENT_EXPORT(SIGNIN_SWITCHES)
@@ -466,7 +467,8 @@ enum class ProfilePickerVariation {
 COMPONENT_EXPORT(SIGNIN_SWITCHES)
 extern const base::FeatureParam<ProfilePickerVariation>
     kProfilePickerTextVariation;
-#endif  // BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX)
+#endif  // BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX) || \
+        // BUILDFLAG(IS_QNX)
 
 COMPONENT_EXPORT(SIGNIN_SWITCHES)
 BASE_DECLARE_FEATURE(kProfilesReordering);
