diff --git chrome/browser/ui/signin/signin_view_controller_delegate.h chrome/browser/ui/signin/signin_view_controller_delegate.h
index 65f2d8c8a4e828bf8f5ad1f07075b5a8e9945949..9ad031aa0d85f2e3252b4a79529ea2b743834258 100644
--- chrome/browser/ui/signin/signin_view_controller_delegate.h
+++ chrome/browser/ui/signin/signin_view_controller_delegate.h
@@ -50,7 +50,8 @@ class SigninViewControllerDelegate {
       SyncConfirmationStyle style,
       bool is_sync_promo);
 
-#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX)
+#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX) || \
+    BUILDFLAG(IS_QNX)
   // Returns a platform-specific SigninViewControllerDelegate instance that
   // displays the modal history sync opt in dialog. The returned object should
   // delete itself when the window it's managing is closed.
@@ -59,7 +60,8 @@ class SigninViewControllerDelegate {
       bool should_close_modal_dialog,
       HistorySyncOptinLaunchContext launch_context,
       HistorySyncOptinHelper::FlowCompletedCallback callback);
-#endif  // BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX)
+#endif  // BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX) || \
+        // BUILDFLAG(IS_QNX)
 
   // Returns a platform-specific SigninViewControllerDelegate instance that
   // displays the modal sign in error dialog. The returned object should delete
diff --git chrome/browser/ui/views/profiles/signin_view_controller_delegate_views.cc chrome/browser/ui/views/profiles/signin_view_controller_delegate_views.cc
index 506fa3457868b196cf8f83e1885b96b7855427ef..07f2d5dd3a928f451bbcbc8f7efc8d731b003256 100644
--- chrome/browser/ui/views/profiles/signin_view_controller_delegate_views.cc
+++ chrome/browser/ui/views/profiles/signin_view_controller_delegate_views.cc
@@ -548,7 +548,8 @@ SigninViewControllerDelegate::CreateSyncConfirmationDelegate(
       /*animate_on_resize=*/true);
 }
 
-#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX)
+#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX) || \
+    BUILDFLAG(IS_QNX)
 // static
 SigninViewControllerDelegate*
 SigninViewControllerDelegate::CreateSyncHistoryOptInDelegate(
@@ -566,7 +567,8 @@ SigninViewControllerDelegate::CreateSyncHistoryOptInDelegate(
       /*wait_for_size=*/true, /*should_show_close_button=*/false,
       /*animate_on_resize=*/true);
 }
-#endif  // BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX)
+#endif  // BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX) || \
+        // BUILDFLAG(IS_QNX)
 
 // static
 SigninViewControllerDelegate*
