diff --git ui/gfx/BUILD.gn ui/gfx/BUILD.gn
index ba4f7988df80d7bb4fabf4b2095a115c7a3f010a..439fb5007c9adc12a88930e813f45f2ff14469ff 100644
--- ui/gfx/BUILD.gn
+++ ui/gfx/BUILD.gn
@@ -231,7 +231,7 @@ component("gfx") {
     deps += [ "//ui/linux:linux_ui" ]
   }
 
-  if (is_linux || is_chromeos) {
+  if (is_linux || is_chromeos || is_qnx) {
     sources += [
       "font_fallback_linux.cc",
       "font_fallback_linux.h",
