diff --git ui/views/BUILD.gn ui/views/BUILD.gn
--- ui/views/BUILD.gn
+++ ui/views/BUILD.gn
@@ -919,7 +919,7 @@ component("views") {
           "//ui/base:ozone_buildflags",
         ]
       }
-      if (is_linux || is_fuchsia) {
+      if (is_linux || is_fuchsia || is_qnx) {
         public += [
           "widget/desktop_aura/desktop_window_tree_host_platform.h",
           "widget/desktop_aura/window_move_client_platform.h",
