diff --git components/update_client/update_query_params.cc components/update_client/update_query_params.cc
index 779caecf3676e..32325bb66e0d0 100644
--- components/update_client/update_query_params.cc
+++ components/update_client/update_query_params.cc
@@ -33,6 +33,8 @@ constexpr std::string_view kOs =
     "android";
 #elif BUILDFLAG(IS_CHROMEOS)
     "cros";
+#elif BUILDFLAG(IS_QNX)
+    "qnx";
 #elif BUILDFLAG(IS_LINUX)
     "linux";
 #elif BUILDFLAG(IS_FUCHSIA)
