diff --git chrome/browser/ui/webui/user_education_internals/user_education_internals_ui.cc chrome/browser/ui/webui/user_education_internals/user_education_internals_ui.cc
index 567b9126003f2..4f35a7f9dd683 100644
--- chrome/browser/ui/webui/user_education_internals/user_education_internals_ui.cc
+++ chrome/browser/ui/webui/user_education_internals/user_education_internals_ui.cc
@@ -33,12 +33,12 @@ UserEducationInternalsUI::UserEducationInternalsUI(content::WebUI* web_ui)
       Profile::FromWebUI(web_ui), chrome::kChromeUIUserEducationInternalsHost);
 
   int32_t version_to_request = CHROME_VERSION_MAJOR;
-#if !BUILDFLAG(IS_CHROMEOS)
+#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX)
   auto* registry = g_browser_process->GetFeatures()->whats_new_registry();
   CHECK(registry);
   version_to_request =
       registry->version_override().value_or(version_to_request);
-#endif  // !BUILDFLAG(IS_CHROMEOS)
+#endif  // BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX)
   source->AddInteger("whatsNewVersionToRequest", version_to_request);
   source->AddInteger("currentChromeVersion", CHROME_VERSION_MAJOR);
 
