diff --git chrome/browser/ui/chrome_pages.cc chrome/browser/ui/chrome_pages.cc
index e05fb064e75b72206b90347cc9925cc1011f0e7e..a8ccc438138edf7e6020797f23cc53a8c2761d13 100644
--- chrome/browser/ui/chrome_pages.cc
+++ chrome/browser/ui/chrome_pages.cc
@@ -700,7 +700,8 @@ void ShowAppManagementPage(Profile* profile,
 }
 #endif  // BUILDFLAG(IS_CHROMEOS)
 
-#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX)
+#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX) || \
+    BUILDFLAG(IS_QNX)
 void ShowWebAppSettingsImpl(Browser* browser,
                             Profile* profile,
                             const std::string& app_id,
diff --git chrome/browser/ui/chrome_pages.h chrome/browser/ui/chrome_pages.h
index 78cbb21ebc5d5e8052da408611459287da697f4b..151f8c6e5b2c5c4eac94690a9b3f211eb3e220ff 100644
--- chrome/browser/ui/chrome_pages.h
+++ chrome/browser/ui/chrome_pages.h
@@ -39,10 +39,9 @@ namespace signin {
 enum class ConsentLevel;
 }  // namespace signin
 
-#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX)
-namespace web_app {
-enum class AppSettingsPageEntryPoint;
-}  // namespace web_app
+#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX) || \
+    BUILDFLAG(IS_QNX)
+#include "chrome/browser/web_applications/web_app_utils.h"
 #endif
 
 class Browser;
@@ -164,7 +163,8 @@ void ShowAppManagementPage(Profile* profile,
                            ash::settings::AppManagementEntryPoint entry_point);
 #endif
 
-#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX)
+#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX) || \
+    BUILDFLAG(IS_QNX)
 // Show chrome://app-settings/<app-id> page.
 void ShowWebAppSettings(Browser* browser,
                         const std::string& app_id,
diff --git chrome/browser/web_applications/web_app_install_params.h chrome/browser/web_applications/web_app_install_params.h
index 618952d89a3a1bd94c3323d0b0e55336b2323696..ee8637c9c617d0c94732325b25a22c5e0bc155f2 100644
--- chrome/browser/web_applications/web_app_install_params.h
+++ chrome/browser/web_applications/web_app_install_params.h
@@ -123,2 +123,3 @@ enum class WebAppInstallFlow {
-#if BUILDFLAG(IS_CHROMEOS)
-  // Perform the `Create Shortcut` flow on CrOS that creates a DIY app.
+#if BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_QNX)
+  // Perform the `Create Shortcut` flow on ChromeOS or QNX that creates a DIY
+  // app.
diff --git chrome/browser/ui/web_applications/web_app_dialog_utils.cc chrome/browser/ui/web_applications/web_app_dialog_utils.cc
index 298f557e0e1743f21278f41ba4992e6486cf7d9d..c3e600679fc2475e31f90789769731e349164534 100644
--- chrome/browser/ui/web_applications/web_app_dialog_utils.cc
+++ chrome/browser/ui/web_applications/web_app_dialog_utils.cc
@@ -105 +105 @@ void OnWebAppInstallShowInstallDialog(
-#if BUILDFLAG(IS_CHROMEOS)
+#if BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_QNX)
@@ -106,0 +107 @@ void OnWebAppInstallShowInstallDialog(
+#if BUILDFLAG(IS_CHROMEOS)
@@ -111 +112 @@ void OnWebAppInstallShowInstallDialog(
-    }
+#endif
@@ -116,0 +118 @@ void OnWebAppInstallShowInstallDialog(
+    }
@@ -201 +203 @@ void CreateWebAppFromCurrentWebContents(Browser* browser,
-#if BUILDFLAG(IS_CHROMEOS)
+#if BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_QNX)
@@ -216 +218 @@ void CreateWebAppFromCurrentWebContents(Browser* browser,
-#if BUILDFLAG(IS_CHROMEOS)
+#if BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_QNX)
