diff --git chrome/browser/ui/web_applications/web_app_dialogs.h chrome/browser/ui/web_applications/web_app_dialogs.h
index 43bec844620110cf108df4d79178fcd13659dc00..fbd58edcf52b025f748b51ce88a9b8cf135c4ace 100644
--- chrome/browser/ui/web_applications/web_app_dialogs.h
+++ chrome/browser/ui/web_applications/web_app_dialogs.h
@@ -25,7 +25,7 @@
 #include "ui/gfx/native_ui_types.h"
 
 static_assert(BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX) ||
-              BUILDFLAG(IS_CHROMEOS));
+              BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_QNX));
 
 class GURL;
 class Profile;
